<!DOCTYPE html> <html lang="en-US"> <head> <meta charset="UTF-8"> <meta http-equiv="X-UA-Compatible" content="IE=Edge"> <title>Named Allele Matcher 101 - PharmCAT</title> <link rel="shortcut icon" href="/favicon.ico" type="image/x-icon"> <link rel="stylesheet" href="/assets/css/just-the-docs-default.css"> <script async src="https://www.googletagmanager.com/gtag/js?id=G-7ZP1HHBWTE"></script> <script> window.dataLayer = window.dataLayer || []; function gtag(){dataLayer.push(arguments);} gtag('js', new Date()); gtag('config', 'G-7ZP1HHBWTE', { 'anonymize_ip': true }); </script> <script type="text/javascript" src="/assets/js/vendor/lunr.min.js"></script> <script type="text/javascript" src="/assets/js/just-the-docs.js"></script> <meta name="viewport" content="width=device-width, initial-scale=1"> <!-- Begin Jekyll SEO tag v2.8.0 --> <title>Named Allele Matcher 101 | PharmCAT</title> <meta name="generator" content="Jekyll v4.2.2" /> <meta property="og:title" content="Named Allele Matcher 101" /> <meta property="og:locale" content="en_US" /> <meta name="description" content="The Pharmacogenomics Clinical Annotation Tool (PharmCAT) is an open-source tool that takes an individual’s genetic data and determines their respective drug-related genotypes and phenotypes. PharmCAT produces a personalized report with genotype-specific expert-reviewed drug prescribing recommendations." /> <meta property="og:description" content="The Pharmacogenomics Clinical Annotation Tool (PharmCAT) is an open-source tool that takes an individual’s genetic data and determines their respective drug-related genotypes and phenotypes. PharmCAT produces a personalized report with genotype-specific expert-reviewed drug prescribing recommendations." /> <meta property="og:site_name" content="PharmCAT" /> <meta property="og:type" content="website" /> <meta name="twitter:card" content="summary" /> <meta property="twitter:title" content="Named Allele Matcher 101" /> <script type="application/ld+json"> {"@context":"https://schema.org","@type":"WebPage","description":"The Pharmacogenomics Clinical Annotation Tool (PharmCAT) is an open-source tool that takes an individual’s genetic data and determines their respective drug-related genotypes and phenotypes. PharmCAT produces a personalized report with genotype-specific expert-reviewed drug prescribing recommendations.","headline":"Named Allele Matcher 101","publisher":{"@type":"Organization","logo":{"@type":"ImageObject","url":"/images/pharmcat_header.png"}},"url":"/methods/NamedAlleleMatcher-101.html"}</script> <!-- End Jekyll SEO tag --> <script async defer data-website-id="714c506c-5f1e-4ea8-af79-fa80c02911c6" src="https://metrics.cpicpgx.org/j.js"></script> <style> h4 { font-size: 14px !important; font-weight: bold; text-transform: none; } h4 code { font-size: 0.9em; font-weight: bold; } h5 code { font-size: .9em; } .main-content ul, .main-content ol { padding-left: 2em; } .main-content dl { grid-template-columns: 16em 1fr; } .main-content dt::after { content: unset; } .altArg { font-size: .8em; font-weight: normal; } footer { position: relative; top: -1.5em; font-size: smaller; } </style> </head> <body> <svg xmlns="http://www.w3.org/2000/svg" style="display: none;"> <symbol id="svg-link" viewBox="0 0 24 24"> <title>Link</title> <svg xmlns="http://www.w3.org/2000/svg" width="24" height="24" viewBox="0 0 24 24" fill="none" stroke="currentColor" stroke-width="2" stroke-linecap="round" stroke-linejoin="round" class="feather feather-link"> <path d="M10 13a5 5 0 0 0 7.54.54l3-3a5 5 0 0 0-7.07-7.07l-1.72 1.71"></path><path d="M14 11a5 5 0 0 0-7.54-.54l-3 3a5 5 0 0 0 7.07 7.07l1.71-1.71"></path> </svg> </symbol> <symbol id="svg-search" viewBox="0 0 24 24"> <title>Search</title> <svg xmlns="http://www.w3.org/2000/svg" width="24" height="24" viewBox="0 0 24 24" fill="none" stroke="currentColor" stroke-width="2" stroke-linecap="round" stroke-linejoin="round" class="feather feather-search"> <circle cx="11" cy="11" r="8"></circle><line x1="21" y1="21" x2="16.65" y2="16.65"></line> </svg> </symbol> <symbol id="svg-menu" viewBox="0 0 24 24"> <title>Menu</title> <svg xmlns="http://www.w3.org/2000/svg" width="24" height="24" viewBox="0 0 24 24" fill="none" stroke="currentColor" stroke-width="2" stroke-linecap="round" stroke-linejoin="round" class="feather feather-menu"> <line x1="3" y1="12" x2="21" y2="12"></line><line x1="3" y1="6" x2="21" y2="6"></line><line x1="3" y1="18" x2="21" y2="18"></line> </svg> </symbol> <symbol id="svg-arrow-right" viewBox="0 0 24 24"> <title>Expand</title> <svg xmlns="http://www.w3.org/2000/svg" width="24" height="24" viewBox="0 0 24 24" fill="none" stroke="currentColor" stroke-width="2" stroke-linecap="round" stroke-linejoin="round" class="feather feather-chevron-right"> <polyline points="9 18 15 12 9 6"></polyline> </svg> </symbol> <symbol id="svg-doc" viewBox="0 0 24 24"> <title>Document</title> <svg xmlns="http://www.w3.org/2000/svg" width="24" height="24" viewBox="0 0 24 24" fill="none" stroke="currentColor" stroke-width="2" stroke-linecap="round" stroke-linejoin="round" class="feather feather-file"> <path d="M13 2H6a2 2 0 0 0-2 2v16a2 2 0 0 0 2 2h12a2 2 0 0 0 2-2V9z"></path><polyline points="13 2 13 9 20 9"></polyline> </svg> </symbol> </svg> <div class="side-bar"> <div class="site-header"> <a href="/" class="site-title lh-tight"> <div class="site-logo"></div> </a> <a href="#" id="menu-button" class="site-button"> <svg viewBox="0 0 24 24" class="icon"><use xlink:href="#svg-menu"></use></svg> </a> </div> <nav role="navigation" aria-label="Main" id="site-nav" class="site-nav"> <ul class="nav-list"><li class="nav-list-item"><a href="/" class="nav-list-link">Home</a></li><li class="nav-list-item active"><a href="#" class="nav-list-expander"><svg viewBox="0 0 24 24"><use xlink:href="#svg-arrow-right"></use></svg></a><a href="/methods/" class="nav-list-link">Methods</a><ul class="nav-list "><li class="nav-list-item "><a href="/methods/Modules.html" class="nav-list-link">PharmCAT Modules</a></li><li class="nav-list-item "><a href="/methods/determining-alleles" class="nav-list-link">Determining Alleles</a></li><li class="nav-list-item active"><a href="/methods/NamedAlleleMatcher-101.html" class="nav-list-link active">Named Allele Matcher 101</a></li><li class="nav-list-item "><a href="/methods/NamedAlleleMatcher-201.html" class="nav-list-link">Named Allele Matcher 201</a></li><li class="nav-list-item "><a href="/methods/gene-definition-exceptions" class="nav-list-link">Gene Definition Exceptions</a></li><li class="nav-list-item "><a href="/methods/versioning/" class="nav-list-link">Versioning</a></li></ul></li><li class="nav-list-item"><a href="#" class="nav-list-expander"><svg viewBox="0 0 24 24"><use xlink:href="#svg-arrow-right"></use></svg></a><a href="/using/" class="nav-list-link">Using PharmCAT</a><ul class="nav-list "><li class="nav-list-item "><a href="/using/Running-PharmCAT.html" class="nav-list-link">Running PharmCAT</a></li><li class="nav-list-item "><a href="/using/VCF-Preprocessor.html" class="nav-list-link">VCF Preprocessor</a></li><li class="nav-list-item "><a href="/using/PharmCAT-in-Docker.html" class="nav-list-link">PharmCAT in Docker</a></li><li class="nav-list-item "><a href="/using/Multi-Sample-Analysis.html" class="nav-list-link">Multi-Sample Analysis</a></li><li class="nav-list-item "><a href="/using/Calling-CYP2D6.html" class="nav-list-link">Calling CYP2D6</a></li></ul></li><li class="nav-list-item"><a href="#" class="nav-list-expander"><svg viewBox="0 0 24 24"><use xlink:href="#svg-arrow-right"></use></svg></a><a href="/specifications/" class="nav-list-link">Specifications</a><ul class="nav-list "><li class="nav-list-item "><a href="/specifications/VCF-Requirements.html" class="nav-list-link">VCF Requirements</a></li><li class="nav-list-item "><a href="/specifications/preparing-vcf" class="nav-list-link">Preparing VCF</a></li><li class="nav-list-item "><a href="/specifications/Outside-Call-Format.html" class="nav-list-link">Outside Call Format</a></li></ul></li><li class="nav-list-item"><a href="/genes-drugs/" class="nav-list-link">Genes & Drugs</a></li><li class="nav-list-item"><a href="/examples/" class="nav-list-link">Examples</a></li><li class="nav-list-item"><a href="/faqs/" class="nav-list-link">FAQs</a></li><li class="nav-list-item"><a href="/disclaimers/" class="nav-list-link">Disclaimers and Other Information</a></li><li class="nav-list-item"><a href="/about-us/" class="nav-list-link">About Us</a></li></ul> </nav> <footer class="site-footer"> This site uses <a href="https://github.com/pmarsceill/just-the-docs">Just the Docs</a>, a documentation theme for Jekyll. </footer> </div> <div class="main" id="top"> <div id="main-header" class="main-header"> <div class="search"> <div class="search-input-wrap"> <input type="text" id="search-input" class="search-input" tabindex="0" placeholder="Search PharmCAT" aria-label="Search PharmCAT" autocomplete="off"> <label for="search-input" class="search-label"><svg viewBox="0 0 24 24" class="search-icon"><use xlink:href="#svg-search"></use></svg></label> </div> <div id="search-results" class="search-results"></div> </div> </div> <div id="main-content-wrap" class="main-content-wrap"> <nav aria-label="Breadcrumb" class="breadcrumb-nav"> <ol class="breadcrumb-nav-list"> <li class="breadcrumb-nav-list-item"><a href="/methods/">Methods</a></li> <li class="breadcrumb-nav-list-item"><span>Named Allele Matcher 101</span></li> </ol> </nav> <div id="main-content" class="main-content" role="main"> <h1 id="named-allele-matcher-101"> <a href="#named-allele-matcher-101" class="anchor-heading" aria-labelledby="named-allele-matcher-101"><svg viewBox="0 0 16 16" aria-hidden="true"><use xlink:href="#svg-link"></use></svg></a> Named Allele Matcher 101 </h1> <p>The basic process:</p> <ol> <li>Read in all named allele definitions from the gene definition table. Each gene has a reference allele defined by the first definition row in the table (eg. <em>*1</em>). By default, any non-reference named allele that does not contain a base call for a given position (eg. blank spots in the definition table) will default to the reference row's base call.</li> <li>Read in sample data (VCF file), ignoring positions that are not used in the gene definition tables.</li> <li>For each gene: <ol> <li>If data is unphased, generate all possible combinations of genotypes for the positions of interest.</li> <li>Attempt to match each combination to a named allele.<sup><a href="#notes">1</a></sup></li> <li>If there are matches, and data is unphased, try to build diplotypes by making sure that the genotype combinations are possible.</li> <li>If diplotypes can be found, they are scored and only the top-scoring diplotype(s) is returned.<sup><a href="#notes">2</a></sup></li> </ol> </li> </ol> <h3 id="scoring"> <a href="#scoring" class="anchor-heading" aria-labelledby="scoring"><svg viewBox="0 0 16 16" aria-hidden="true"><use xlink:href="#svg-link"></use></svg></a> Scoring </h3> <p>Each named allele is given a score based on the number of variant positions used to define the allele (non-blank cells in that row). This means that the reference allele will always have the maximum score because all positions are defined for that allele.</p> <p>Take a look at this sample gene definition table:</p> <div class="table-wrapper"><table> <thead> <tr> <th> </th> <th>rs1</th> <th>rs2</th> <th>rs3</th> <th>rs4</th> <th>rs5</th> <th>score</th> </tr> </thead> <tbody> <tr> <td>*1</td> <td>C</td> <td>C</td> <td>T</td> <td>G</td> <td>A</td> <td>5</td> </tr> <tr> <td>*2</td> <td>T</td> <td>T</td> <td> </td> <td>A</td> <td> </td> <td>3</td> </tr> </tbody> </table></div> <p>Since the gene definition table contains 5 positions, the reference allele, <em>*1</em>, gets a score of 5 while <em>*2</em> only has 3 positions defined and gets a score of 3. <sup><a href="#notes">3</a></sup></p> <p>A diplotype's score is the combined score of its component named alleles. A <em>*1</em>/<em>*2</em> from the example above would have a score of 8.</p> <h3 id="missing-positions"> <a href="#missing-positions" class="anchor-heading" aria-labelledby="missing-positions"><svg viewBox="0 0 16 16" aria-hidden="true"><use xlink:href="#svg-link"></use></svg></a> Missing Positions </h3> <p>If the sample data has missing positions that are required by a named allele definition, the position will be dropped from consideration.</p> <p>This is the only reason the score for a diplotype might be different between two samples.</p> <h3 id="examples"> <a href="#examples" class="anchor-heading" aria-labelledby="examples"><svg viewBox="0 0 16 16" aria-hidden="true"><use xlink:href="#svg-link"></use></svg></a> Examples </h3> <p>Using the following gene definition table:</p> <div class="table-wrapper"><table> <thead> <tr> <th> </th> <th>rs1</th> <th>rs2</th> <th>rs3</th> <th>rs4</th> <th>rs5</th> <th>score</th> </tr> </thead> <tbody> <tr> <td>*1</td> <td>C</td> <td>C</td> <td>T</td> <td>G</td> <td>A</td> <td>5</td> </tr> <tr> <td>*2</td> <td>T</td> <td> </td> <td> </td> <td>A</td> <td> </td> <td>2</td> </tr> <tr> <td>*3</td> <td> </td> <td> </td> <td> </td> <td>A</td> <td> </td> <td>1</td> </tr> <tr> <td>*4</td> <td> </td> <td>T</td> <td> </td> <td> </td> <td> </td> <td>1</td> </tr> <tr> <td>*5</td> <td>T</td> <td> </td> <td> </td> <td> </td> <td> </td> <td>1</td> </tr> </tbody> </table></div> <p>And the following (unphased) sample data:</p> <div class="table-wrapper"><table> <thead> <tr> <th>rs1</th> <th>rs2</th> <th>rs3</th> <th>rs4</th> <th>rs5</th> </tr> </thead> <tbody> <tr> <td>C/T</td> <td>C/C</td> <td>T/T</td> <td>A/G</td> <td>A/A</td> </tr> </tbody> </table></div> <p>The potential permutations of those genotypes will match <em>*1</em>, <em>*2</em>, <em>*3</em> and <em>*5</em>.</p> <p>From that, plausible diplotypes are:</p> <div class="table-wrapper"><table> <thead> <tr> <th>Diplotype</th> <th>Score</th> </tr> </thead> <tbody> <tr> <td><em>1/</em>2</td> <td>5 + 2 = 7</td> </tr> <tr> <td><em>3/</em>5</td> <td>1 + 1 = 2</td> </tr> </tbody> </table></div> <p>Which results in <em>*1</em>/<em>*2</em> being returned.</p> <p>Note that <em>*1</em>/<em>*3</em> is not a plausible diplotype because one chromosome must have a <code class="language-plaintext highlighter-rouge">C</code> and the other must have a <code class="language-plaintext highlighter-rouge">T</code> at position rs1. They can't both be <code class="language-plaintext highlighter-rouge">C</code>. Similarly, <em>*2</em>/<em>*3</em> is not a plausible diplotype either because it cannot be homozygous at rs4.</p> <h4 id="missing-rs5"> <a href="#missing-rs5" class="anchor-heading" aria-labelledby="missing-rs5"><svg viewBox="0 0 16 16" aria-hidden="true"><use xlink:href="#svg-link"></use></svg></a> Missing rs5 </h4> <p>If we use the same gene definition table as the example above and the following (unphased) sample data, <em>with no data available for rs5</em>:</p> <div class="table-wrapper"><table> <thead> <tr> <th>rs1</th> <th>rs2</th> <th>rs3</th> <th>rs4</th> <th>rs5</th> </tr> </thead> <tbody> <tr> <td>C/T</td> <td>C/C</td> <td>T/T</td> <td>A/G</td> <td> </td> </tr> </tbody> </table></div> <p>The results would be the same, except the scores would be different:</p> <div class="table-wrapper"><table> <thead> <tr> <th>Diplotype</th> <th>Score</th> </tr> </thead> <tbody> <tr> <td><em>1/</em>2</td> <td>4 + 2 = 6</td> </tr> <tr> <td><em>3/</em>5</td> <td>1 + 1 = 2</td> </tr> </tbody> </table></div> <h4 id="missing-rs1"> <a href="#missing-rs1" class="anchor-heading" aria-labelledby="missing-rs1"><svg viewBox="0 0 16 16" aria-hidden="true"><use xlink:href="#svg-link"></use></svg></a> Missing rs1 </h4> <p>If we use the same gene definition table as the example above and the following (unphased) sample data, <em>with no data available for rs1</em>:</p> <div class="table-wrapper"><table> <thead> <tr> <th>rs1</th> <th>rs2</th> <th>rs3</th> <th>rs4</th> <th>rs5</th> </tr> </thead> <tbody> <tr> <td> </td> <td>C/C</td> <td>T/T</td> <td>A/G</td> <td>A/A</td> </tr> </tbody> </table></div> <p>Then the results would be different:</p> <div class="table-wrapper"><table> <thead> <tr> <th>Diplotype</th> <th>Score</th> </tr> </thead> <tbody> <tr> <td><em>1/</em>2</td> <td>4 + 1 = 5</td> </tr> <tr> <td><em>1/</em>3</td> <td>4 + 1 = 5</td> </tr> </tbody> </table></div> <p>As such, <em>*1</em>/<em>*2</em> and <em>*1</em>/<em>*3</em> would be returned.</p> <p>Note that *5 could never be matched in this scenario because it's definining allele is missing.</p> <h3 id="notes"> <a href="#notes" class="anchor-heading" aria-labelledby="notes"><svg viewBox="0 0 16 16" aria-hidden="true"><use xlink:href="#svg-link"></use></svg></a> Notes </h3> <p><strong>1</strong>: If sample data is not phased and we do not assume the reference for missing positions in the definition, it is possible to have multiple matches for a single named allele.</p> <p><strong>2</strong>: This behavior can be modified to return all potential diplotype matches.</p> <p><strong>3</strong>: This score is the same irregardless of whether we assume the reference for missing positions.</p> <hr> <footer> <div> PharmCAT is managed at Stanford University & University of Pennsylvania (NHGRI U24HG010862). </div> </footer> </div> </div> <div class="search-overlay"></div> </div> </body> </html>
