<!DOCTYPE html> <html lang="en-US"> <head> <meta charset="UTF-8"> <meta http-equiv="X-UA-Compatible" content="IE=Edge"> <title>Preparing VCF - PharmCAT</title> <link rel="shortcut icon" href="/favicon.ico" type="image/x-icon"> <link rel="stylesheet" href="/assets/css/just-the-docs-default.css"> <script async src="https://www.googletagmanager.com/gtag/js?id=G-7ZP1HHBWTE"></script> <script> window.dataLayer = window.dataLayer || []; function gtag(){dataLayer.push(arguments);} gtag('js', new Date()); gtag('config', 'G-7ZP1HHBWTE', { 'anonymize_ip': true }); </script> <script type="text/javascript" src="/assets/js/vendor/lunr.min.js"></script> <script type="text/javascript" src="/assets/js/just-the-docs.js"></script> <meta name="viewport" content="width=device-width, initial-scale=1"> <!-- Begin Jekyll SEO tag v2.8.0 --> <title>Preparing VCF | PharmCAT</title> <meta name="generator" content="Jekyll v4.2.2" /> <meta property="og:title" content="Preparing VCF" /> <meta property="og:locale" content="en_US" /> <meta name="description" content="The Pharmacogenomics Clinical Annotation Tool (PharmCAT) is an open-source tool that takes an individual’s genetic data and determines their respective drug-related genotypes and phenotypes. PharmCAT produces a personalized report with genotype-specific expert-reviewed drug prescribing recommendations." /> <meta property="og:description" content="The Pharmacogenomics Clinical Annotation Tool (PharmCAT) is an open-source tool that takes an individual’s genetic data and determines their respective drug-related genotypes and phenotypes. PharmCAT produces a personalized report with genotype-specific expert-reviewed drug prescribing recommendations." /> <meta property="og:site_name" content="PharmCAT" /> <meta property="og:type" content="website" /> <meta name="twitter:card" content="summary" /> <meta property="twitter:title" content="Preparing VCF" /> <script type="application/ld+json"> {"@context":"https://schema.org","@type":"WebPage","description":"The Pharmacogenomics Clinical Annotation Tool (PharmCAT) is an open-source tool that takes an individual’s genetic data and determines their respective drug-related genotypes and phenotypes. PharmCAT produces a personalized report with genotype-specific expert-reviewed drug prescribing recommendations.","headline":"Preparing VCF","publisher":{"@type":"Organization","logo":{"@type":"ImageObject","url":"/images/pharmcat_header.png"}},"url":"/specifications/preparing-vcf"}</script> <!-- End Jekyll SEO tag --> <script async defer data-website-id="714c506c-5f1e-4ea8-af79-fa80c02911c6" src="https://metrics.cpicpgx.org/j.js"></script> <style> h4 { font-size: 14px !important; font-weight: bold; text-transform: none; } h4 code { font-size: 0.9em; font-weight: bold; } h5 code { font-size: .9em; } .main-content ul, .main-content ol { padding-left: 2em; } .main-content dl { grid-template-columns: 16em 1fr; } .main-content dt::after { content: unset; } .altArg { font-size: .8em; font-weight: normal; } footer { position: relative; top: -1.5em; font-size: smaller; } </style> </head> <body> <svg xmlns="http://www.w3.org/2000/svg" style="display: none;"> <symbol id="svg-link" viewBox="0 0 24 24"> <title>Link</title> <svg xmlns="http://www.w3.org/2000/svg" width="24" height="24" viewBox="0 0 24 24" fill="none" stroke="currentColor" stroke-width="2" stroke-linecap="round" stroke-linejoin="round" class="feather feather-link"> <path d="M10 13a5 5 0 0 0 7.54.54l3-3a5 5 0 0 0-7.07-7.07l-1.72 1.71"></path><path d="M14 11a5 5 0 0 0-7.54-.54l-3 3a5 5 0 0 0 7.07 7.07l1.71-1.71"></path> </svg> </symbol> <symbol id="svg-search" viewBox="0 0 24 24"> <title>Search</title> <svg xmlns="http://www.w3.org/2000/svg" width="24" height="24" viewBox="0 0 24 24" fill="none" stroke="currentColor" stroke-width="2" stroke-linecap="round" stroke-linejoin="round" class="feather feather-search"> <circle cx="11" cy="11" r="8"></circle><line x1="21" y1="21" x2="16.65" y2="16.65"></line> </svg> </symbol> <symbol id="svg-menu" viewBox="0 0 24 24"> <title>Menu</title> <svg xmlns="http://www.w3.org/2000/svg" width="24" height="24" viewBox="0 0 24 24" fill="none" stroke="currentColor" stroke-width="2" stroke-linecap="round" stroke-linejoin="round" class="feather feather-menu"> <line x1="3" y1="12" x2="21" y2="12"></line><line x1="3" y1="6" x2="21" y2="6"></line><line x1="3" y1="18" x2="21" y2="18"></line> </svg> </symbol> <symbol id="svg-arrow-right" viewBox="0 0 24 24"> <title>Expand</title> <svg xmlns="http://www.w3.org/2000/svg" width="24" height="24" viewBox="0 0 24 24" fill="none" stroke="currentColor" stroke-width="2" stroke-linecap="round" stroke-linejoin="round" class="feather feather-chevron-right"> <polyline points="9 18 15 12 9 6"></polyline> </svg> </symbol> <symbol id="svg-doc" viewBox="0 0 24 24"> <title>Document</title> <svg xmlns="http://www.w3.org/2000/svg" width="24" height="24" viewBox="0 0 24 24" fill="none" stroke="currentColor" stroke-width="2" stroke-linecap="round" stroke-linejoin="round" class="feather feather-file"> <path d="M13 2H6a2 2 0 0 0-2 2v16a2 2 0 0 0 2 2h12a2 2 0 0 0 2-2V9z"></path><polyline points="13 2 13 9 20 9"></polyline> </svg> </symbol> </svg> <div class="side-bar"> <div class="site-header"> <a href="/" class="site-title lh-tight"> <div class="site-logo"></div> </a> <a href="#" id="menu-button" class="site-button"> <svg viewBox="0 0 24 24" class="icon"><use xlink:href="#svg-menu"></use></svg> </a> </div> <nav role="navigation" aria-label="Main" id="site-nav" class="site-nav"> <ul class="nav-list"><li class="nav-list-item"><a href="/" class="nav-list-link">Home</a></li><li class="nav-list-item"><a href="#" class="nav-list-expander"><svg viewBox="0 0 24 24"><use xlink:href="#svg-arrow-right"></use></svg></a><a href="/methods/" class="nav-list-link">Methods</a><ul class="nav-list "><li class="nav-list-item "><a href="/methods/Modules.html" class="nav-list-link">PharmCAT Modules</a></li><li class="nav-list-item "><a href="/methods/determining-alleles" class="nav-list-link">Determining Alleles</a></li><li class="nav-list-item "><a href="/methods/NamedAlleleMatcher-101.html" class="nav-list-link">Named Allele Matcher 101</a></li><li class="nav-list-item "><a href="/methods/NamedAlleleMatcher-201.html" class="nav-list-link">Named Allele Matcher 201</a></li><li class="nav-list-item "><a href="/methods/gene-definition-exceptions" class="nav-list-link">Gene Definition Exceptions</a></li><li class="nav-list-item "><a href="/methods/versioning/" class="nav-list-link">Versioning</a></li></ul></li><li class="nav-list-item"><a href="#" class="nav-list-expander"><svg viewBox="0 0 24 24"><use xlink:href="#svg-arrow-right"></use></svg></a><a href="/using/" class="nav-list-link">Using PharmCAT</a><ul class="nav-list "><li class="nav-list-item "><a href="/using/Running-PharmCAT.html" class="nav-list-link">Running PharmCAT</a></li><li class="nav-list-item "><a href="/using/VCF-Preprocessor.html" class="nav-list-link">VCF Preprocessor</a></li><li class="nav-list-item "><a href="/using/PharmCAT-in-Docker.html" class="nav-list-link">PharmCAT in Docker</a></li><li class="nav-list-item "><a href="/using/Multi-Sample-Analysis.html" class="nav-list-link">Multi-Sample Analysis</a></li><li class="nav-list-item "><a href="/using/Calling-CYP2D6.html" class="nav-list-link">Calling CYP2D6</a></li></ul></li><li class="nav-list-item active"><a href="#" class="nav-list-expander"><svg viewBox="0 0 24 24"><use xlink:href="#svg-arrow-right"></use></svg></a><a href="/specifications/" class="nav-list-link">Specifications</a><ul class="nav-list "><li class="nav-list-item "><a href="/specifications/VCF-Requirements.html" class="nav-list-link">VCF Requirements</a></li><li class="nav-list-item active"><a href="/specifications/preparing-vcf" class="nav-list-link active">Preparing VCF</a></li><li class="nav-list-item "><a href="/specifications/Outside-Call-Format.html" class="nav-list-link">Outside Call Format</a></li></ul></li><li class="nav-list-item"><a href="/genes-drugs/" class="nav-list-link">Genes & Drugs</a></li><li class="nav-list-item"><a href="/examples/" class="nav-list-link">Examples</a></li><li class="nav-list-item"><a href="/faqs/" class="nav-list-link">FAQs</a></li><li class="nav-list-item"><a href="/disclaimers/" class="nav-list-link">Disclaimers and Other Information</a></li><li class="nav-list-item"><a href="/about-us/" class="nav-list-link">About Us</a></li></ul> </nav> <footer class="site-footer"> This site uses <a href="https://github.com/pmarsceill/just-the-docs">Just the Docs</a>, a documentation theme for Jekyll. </footer> </div> <div class="main" id="top"> <div id="main-header" class="main-header"> <div class="search"> <div class="search-input-wrap"> <input type="text" id="search-input" class="search-input" tabindex="0" placeholder="Search PharmCAT" aria-label="Search PharmCAT" autocomplete="off"> <label for="search-input" class="search-label"><svg viewBox="0 0 24 24" class="search-icon"><use xlink:href="#svg-search"></use></svg></label> </div> <div id="search-results" class="search-results"></div> </div> </div> <div id="main-content-wrap" class="main-content-wrap"> <nav aria-label="Breadcrumb" class="breadcrumb-nav"> <ol class="breadcrumb-nav-list"> <li class="breadcrumb-nav-list-item"><a href="/specifications/">Specifications</a></li> <li class="breadcrumb-nav-list-item"><span>Preparing VCF</span></li> </ol> </nav> <div id="main-content" class="main-content" role="main"> <h1 class="no_toc" id="preparing-vcf"> <a href="#preparing-vcf" class="anchor-heading" aria-labelledby="preparing-vcf"><svg viewBox="0 0 16 16" aria-hidden="true"><use xlink:href="#svg-link"></use></svg></a> Preparing VCF </h1> <p>This document explores the reasoning behind <a href="/specifications/VCF-Requirements">PharmCAT's VCF requirements</a> and some specific examples of ways to fulfill them if you plan on preparing your VCF files yourself.</p> <p>If you do not intend to prepare your VCF files yourself, we highly recommend you use PharmCAT's <a href="/using/VCF-Preprocessor">VCF preprocessor</a>.</p><hr /> <h2 class="no_toc text-delta" id="table-of-contents"> <a href="#table-of-contents" class="anchor-heading" aria-labelledby="table-of-contents"><svg viewBox="0 0 16 16" aria-hidden="true"><use xlink:href="#svg-link"></use></svg></a> Table of contents </h2> <ul id="markdown-toc"> <li><a href="#must-be-aligned-to-grch38-assembly" id="markdown-toc-must-be-aligned-to-grch38-assembly">Must be aligned to GRCh38 Assembly</a></li> <li><a href="#must-have-all-allele-defining-positions" id="markdown-toc-must-have-all-allele-defining-positions">Must have all allele-defining positions</a></li> <li><a href="#must-normalize-variant-representation" id="markdown-toc-must-normalize-variant-representation">Must normalize variant representation</a></li> <li><a href="#the-chrom-field-must-be-in-the-format-chr" id="markdown-toc-the-chrom-field-must-be-in-the-format-chr">The <code class="language-plaintext highlighter-rouge">CHROM</code> field must be in the format <strong>chr##</strong></a></li> <li><a href="#the-qual-and-filter-columns-are-not-interpreted" id="markdown-toc-the-qual-and-filter-columns-are-not-interpreted">The <code class="language-plaintext highlighter-rouge">QUAL</code> and <code class="language-plaintext highlighter-rouge">FILTER</code> columns are <strong>not interpreted</strong></a></li> </ul><hr /> <h2 id="must-be-aligned-to-grch38-assembly"> <a href="#must-be-aligned-to-grch38-assembly" class="anchor-heading" aria-labelledby="must-be-aligned-to-grch38-assembly"><svg viewBox="0 0 16 16" aria-hidden="true"><use xlink:href="#svg-link"></use></svg></a> Must be aligned to GRCh38 Assembly </h2> <p>PharmCAT requires VCF files aligned to GRCh38.</p> <p>We recommend aligning to GRCh38 from the very beginning - for example, see our <a href="https://github.com/PharmGKB/pgkb-ngs-pipeline">pharmacogenomics NGS pipeline</a>.</p> <p>While you can translate positions between assemblies, this can easily introduce errors into your data. For example, LiftOver can be problematic due to ambiguity in positions. In addition, many tools won't update the genotype.</p> <h3 class="no_toc" id="known-issue-with-remapping"> <a href="#known-issue-with-remapping" class="anchor-heading" aria-labelledby="known-issue-with-remapping"><svg viewBox="0 0 16 16" aria-hidden="true"><use xlink:href="#svg-link"></use></svg></a> Known issue with remapping </h3> <p>If you use <a href="http://crossmap.sourceforge.net/">CrossMap</a> it will update the position and reference nucleotides. However, if the reference is now the same as the alternate, it will remove the variant from the file.</p> <p>Likewise, <a href="http://crossmap.sourceforge.net/">NCBI Remap</a> will not update genotypes, so even if the updated position means the variant should now be called as 0/0 it will not be updated from 1/1. Presently there are only a few positions where this matters, but this will need to be very carefully checked.</p> <p>Another issue with NCBI remap is that some alternates can be lost following remap. For instance:</p> <div class="language-text highlighter-rouge"><div class="highlight"><pre class="highlight"><code> Pre LiftOver: 2	234668879	rs57191451	CAT	C,CATAT,CATATAT	0/3
Post LiftOver: 2	233760233	rs57191451	CAT	CATAT,CATATAT	0/3
</code></pre></div></div> <p>We currently do not have a fix for this (NCBI has been contacted), so again you will have to check these sites carefully by hand.</p> <h3 class="no_toc" id="example-liftover-using-the-gatk"> <a href="#example-liftover-using-the-gatk" class="anchor-heading" aria-labelledby="example-liftover-using-the-gatk"><svg viewBox="0 0 16 16" aria-hidden="true"><use xlink:href="#svg-link"></use></svg></a> Example: LiftOver using the GATK </h3> <p>This is a GRCh37-toGRCh38 LiftOver example which uses the GATK LiftoverVcf tool on the UK Biobank genotype data in the GRCh37 coordinates. The GATK LiftoverVcf tool can update genomic coordinates. If the reference allele and alternate allele are swapped in the new genome build for a single nucleotide polymorphism (SNP) locus, the GATK LiftoverVcf can reverse-complement the SNP, update the relevant genotypes (GT), and correct AF-like INFO fields. Due to the length of the example, we host the <a href="https://docs.google.com/document/d/15rxe0iG2kruEWsvBCLyNGof-YRo5T10zuQiBJkUbyJ0/edit?usp=sharing">example with detailed explanations on Google Drive</a>.</p> <p>Note: this example is not meant to be a comprehensive documentation of solutions to all LiftOver issues. LiftOver may require additional data cleaning or preparation steps that are specific to your genomic data.</p> <h2 id="must-have-all-allele-defining-positions"> <a href="#must-have-all-allele-defining-positions" class="anchor-heading" aria-labelledby="must-have-all-allele-defining-positions"><svg viewBox="0 0 16 16" aria-hidden="true"><use xlink:href="#svg-link"></use></svg></a> Must have all allele-defining positions </h2> <p>All positions that are used to define alleles in PharmCAT must be present in the VCF file you want to run through PharmCAT, <em>even if they are 0/0 (reference) or ./. (missing)</em>. This is different from a typical VCF file which usually only contains variant sites.</p> <p>PharmCAT needs this level specificity because reference and missing alleles are interpreted differently by the allele matcher. When positions are missing from the input PharmCAT does make assumptions about whether this is because they are reference or missing.</p> <p>Missing positions can be added in the following way using GATK to 'EMIT_ALL_ACTIVE_SITES':</p> <div class="language-console highlighter-rouge"><div class="highlight"><pre class="highlight"><code><span class="gp">#</span><span class="w"> </span>gatk <span class="nt">--java-options</span> <span class="s2">"-Xmx4g"</span> HaplotypeCaller <span class="se">\</span>
<span class="go">     -R grc38.reference.fasta -I input.bam -O output.vcf \
     -L pharmcat_positions.vcf -ip 20 --output-mode EMIT_ALL_ACTIVE_SITES
</span></code></pre></div></div> <p>The <code class="language-plaintext highlighter-rouge">pharmcat_positions.vcf</code> file is available on the <a href="https://github.com/PharmGKB/PharmCAT/releases">PharmCAT release page</a> on GitHub.</p> <p>Please refer to the <a href="https://gatk.broadinstitute.org/hc/en-us/articles/360037225632-HaplotypeCaller">HaplotypeCaller</a> documentation for details and tuning options.</p> <h2 id="must-normalize-variant-representation"> <a href="#must-normalize-variant-representation" class="anchor-heading" aria-labelledby="must-normalize-variant-representation"><svg viewBox="0 0 16 16" aria-hidden="true"><use xlink:href="#svg-link"></use></svg></a> Must normalize variant representation </h2> <p>Variant representation is an on-going problem in NGS (<a href="https://macarthurlab.org/2014/04/28/converting-genetic-variants-to-their-minimal-representation/">related article</a>). For example, the following vcf lines all specify the same variant with different formats:</p> <div class="language-text highlighter-rouge"><div class="highlight"><pre class="highlight"><code>chr7    117548628    .    GTTTTTTTA    GTTTTTA    .    PASS    CFTR:5T    GT    0/1
chr7    117548628    .    GTT     G    .    PASS    CFTR:5T    GT    0/1
chr7    117548628    .    G    .    .    PASS    CFTR:5T    GT    0/1
chr7    117548628    .    G(T)7A    G(T)5A    .    PASS    CFTR:5T    GT    0/1
</code></pre></div></div> <p>Different NGS pipelines, the way files are created (for instance if a multisample file is split), and post-processing software tools all lead to these differences. As PharmCAT is directly matching these strings to what is in the definition files this can cause problems. For example, PharmCAT expects to find deletions as the ref="ATCT" alt="A", rather than ref="TCT" alt=".". Therefore you will need to replace all the deletions within in the file. If the ref is a single letter it means no variant was found, so it's safe to replace it with the appropriate nucleotide string.</p> <p>To avoid ambiguity in variant representation, PharmCAT is using a parsimonious, left-aligned variant representation format (as discussed in <a href="https://doi.org/10.1093/bioinformatics/btv112">Unified Representation of Genetic Variants</a> by Tan, Abecasis, and Kang).</p> <p>We recommend performing this normalization with <a href="http://samtools.github.io/bcftools/bcftools.html">bcftools</a>:</p> <div class="language-console highlighter-rouge"><div class="highlight"><pre class="highlight"><code><span class="gp">#</span><span class="w"> </span>bcftools norm <span class="nt">-m</span>+ <span class="nt">-c</span> ws <span class="nt">-Oz</span> <span class="nt">-o</span> output.vcf <span class="nt">-f</span> grc38.reference.fasta input.vcf
</code></pre></div></div> <ul> <li><code class="language-plaintext highlighter-rouge">-m+</code> joins biallelic sites into multiallelic records (+)</li> <li><code class="language-plaintext highlighter-rouge">-f &lt;ref_seq_fasta&gt;</code> is the reference sequence. This flag is required for normalization</li> <li><code class="language-plaintext highlighter-rouge">-c ws</code> when incorrect or missing REF allele is encountered, warn (w) and set/fix(s) bad sites</li> </ul> <p>Please consult the <a href="http://samtools.github.io/bcftools/bcftools.html">bcftools documentation</a> for details.</p> <p>Alternatively, you can use PharmCAT's <a href="/using/VCF-Preprocessor">VCF preprocessor</a>, which also relies on bcftools for this.</p> <p>It is highly recommended that you always check the output files from these tools manually to make sure the correct format normalizations have been made.</p> <h2 id="the-chrom-field-must-be-in-the-format-chr"> <a href="#the-chrom-field-must-be-in-the-format-chr" class="anchor-heading" aria-labelledby="the-chrom-field-must-be-in-the-format-chr"><svg viewBox="0 0 16 16" aria-hidden="true"><use xlink:href="#svg-link"></use></svg></a> The <code class="language-plaintext highlighter-rouge">CHROM</code> field must be in the format <strong>chr##</strong> </h2> <p>PharmCAT expects the <code class="language-plaintext highlighter-rouge">CHROM</code> field to have entries that begin with "chr" (e.g. <code class="language-plaintext highlighter-rouge">chr1</code> instead of just <code class="language-plaintext highlighter-rouge">1</code>).</p> <p><a href="/using/VCF-Preprocessor">PharmCAT's VCF preprocessor</a> takes care of this issue by automatically detecting and updating the <code class="language-plaintext highlighter-rouge">CHROM</code> field format.</p> <p>Alternatively,<code class="language-plaintext highlighter-rouge">CHROM</code> field format can be updated using the following command:</p> <div class="language-console highlighter-rouge"><div class="highlight"><pre class="highlight"><code><span class="gp">#</span><span class="w"> </span>perl <span class="nt">-pe</span> <span class="s1">'/^((?!^chr).)*$/ &amp;&amp; s/^([^#])/chr$1/gsi'</span> merged_output.vcf <span class="o">&gt;</span> merged_output.chrfixed.vcf
</code></pre></div></div> <h2 id="the-qual-and-filter-columns-are-not-interpreted"> <a href="#the-qual-and-filter-columns-are-not-interpreted" class="anchor-heading" aria-labelledby="the-qual-and-filter-columns-are-not-interpreted"><svg viewBox="0 0 16 16" aria-hidden="true"><use xlink:href="#svg-link"></use></svg></a> The <code class="language-plaintext highlighter-rouge">QUAL</code> and <code class="language-plaintext highlighter-rouge">FILTER</code> columns are <strong>not interpreted</strong> </h2> <p>PharmCAT considers all variants in your file, even if they fail filters.</p> <p>If you use a tool like VSQR you can use the following Perl one-liner to change the genotype to 0/0 if necessary:</p> <div class="language-console highlighter-rouge"><div class="highlight"><pre class="highlight"><code><span class="gp">#</span><span class="w"> </span>perl <span class="nt">-pe</span> <span class="s1">'/^((?!PASS).)*$/ &amp;&amp; /^((?!0\/0).)*$/ &amp;&amp; /^((?!0\|0).)*$/ &amp;&amp; s/[0-9][\/|][0-9]/0|0/'</span> merged_output.vcf <span class="o">&gt;</span> merged_output_pass.vcf
</code></pre></div></div> <hr> <footer> <div> PharmCAT is managed at Stanford University & University of Pennsylvania (NHGRI U24HG010862). </div> </footer> </div> </div> <div class="search-overlay"></div> </div> </body> </html>
