<!DOCTYPE html> <html lang="en-US"> <head> <meta charset="UTF-8"> <meta http-equiv="X-UA-Compatible" content="IE=Edge"> <title>Calling CYP2D6 - PharmCAT</title> <link rel="shortcut icon" href="/favicon.ico" type="image/x-icon"> <link rel="stylesheet" href="/assets/css/just-the-docs-default.css"> <script async src="https://www.googletagmanager.com/gtag/js?id=G-7ZP1HHBWTE"></script> <script> window.dataLayer = window.dataLayer || []; function gtag(){dataLayer.push(arguments);} gtag('js', new Date()); gtag('config', 'G-7ZP1HHBWTE', { 'anonymize_ip': true }); </script> <script type="text/javascript" src="/assets/js/vendor/lunr.min.js"></script> <script type="text/javascript" src="/assets/js/just-the-docs.js"></script> <meta name="viewport" content="width=device-width, initial-scale=1"> <!-- Begin Jekyll SEO tag v2.8.0 --> <title>Calling CYP2D6 | PharmCAT</title> <meta name="generator" content="Jekyll v4.2.2" /> <meta property="og:title" content="Calling CYP2D6" /> <meta property="og:locale" content="en_US" /> <meta name="description" content="The Pharmacogenomics Clinical Annotation Tool (PharmCAT) is an open-source tool that takes an individual’s genetic data and determines their respective drug-related genotypes and phenotypes. PharmCAT produces a personalized report with genotype-specific expert-reviewed drug prescribing recommendations." /> <meta property="og:description" content="The Pharmacogenomics Clinical Annotation Tool (PharmCAT) is an open-source tool that takes an individual’s genetic data and determines their respective drug-related genotypes and phenotypes. PharmCAT produces a personalized report with genotype-specific expert-reviewed drug prescribing recommendations." /> <meta property="og:site_name" content="PharmCAT" /> <meta property="og:type" content="website" /> <meta name="twitter:card" content="summary" /> <meta property="twitter:title" content="Calling CYP2D6" /> <script type="application/ld+json"> {"@context":"https://schema.org","@type":"WebPage","description":"The Pharmacogenomics Clinical Annotation Tool (PharmCAT) is an open-source tool that takes an individual’s genetic data and determines their respective drug-related genotypes and phenotypes. PharmCAT produces a personalized report with genotype-specific expert-reviewed drug prescribing recommendations.","headline":"Calling CYP2D6","publisher":{"@type":"Organization","logo":{"@type":"ImageObject","url":"/images/pharmcat_header.png"}},"url":"/using/Calling-CYP2D6.html"}</script> <!-- End Jekyll SEO tag --> <script async defer data-website-id="714c506c-5f1e-4ea8-af79-fa80c02911c6" src="https://metrics.cpicpgx.org/j.js"></script> <style> h4 { font-size: 14px !important; font-weight: bold; text-transform: none; } h4 code { font-size: 0.9em; font-weight: bold; } h5 code { font-size: .9em; } .main-content ul, .main-content ol { padding-left: 2em; } .main-content dl { grid-template-columns: 16em 1fr; } .main-content dt::after { content: unset; } .altArg { font-size: .8em; font-weight: normal; } footer { position: relative; top: -1.5em; font-size: smaller; } </style> </head> <body> <svg xmlns="http://www.w3.org/2000/svg" style="display: none;"> <symbol id="svg-link" viewBox="0 0 24 24"> <title>Link</title> <svg xmlns="http://www.w3.org/2000/svg" width="24" height="24" viewBox="0 0 24 24" fill="none" stroke="currentColor" stroke-width="2" stroke-linecap="round" stroke-linejoin="round" class="feather feather-link"> <path d="M10 13a5 5 0 0 0 7.54.54l3-3a5 5 0 0 0-7.07-7.07l-1.72 1.71"></path><path d="M14 11a5 5 0 0 0-7.54-.54l-3 3a5 5 0 0 0 7.07 7.07l1.71-1.71"></path> </svg> </symbol> <symbol id="svg-search" viewBox="0 0 24 24"> <title>Search</title> <svg xmlns="http://www.w3.org/2000/svg" width="24" height="24" viewBox="0 0 24 24" fill="none" stroke="currentColor" stroke-width="2" stroke-linecap="round" stroke-linejoin="round" class="feather feather-search"> <circle cx="11" cy="11" r="8"></circle><line x1="21" y1="21" x2="16.65" y2="16.65"></line> </svg> </symbol> <symbol id="svg-menu" viewBox="0 0 24 24"> <title>Menu</title> <svg xmlns="http://www.w3.org/2000/svg" width="24" height="24" viewBox="0 0 24 24" fill="none" stroke="currentColor" stroke-width="2" stroke-linecap="round" stroke-linejoin="round" class="feather feather-menu"> <line x1="3" y1="12" x2="21" y2="12"></line><line x1="3" y1="6" x2="21" y2="6"></line><line x1="3" y1="18" x2="21" y2="18"></line> </svg> </symbol> <symbol id="svg-arrow-right" viewBox="0 0 24 24"> <title>Expand</title> <svg xmlns="http://www.w3.org/2000/svg" width="24" height="24" viewBox="0 0 24 24" fill="none" stroke="currentColor" stroke-width="2" stroke-linecap="round" stroke-linejoin="round" class="feather feather-chevron-right"> <polyline points="9 18 15 12 9 6"></polyline> </svg> </symbol> <symbol id="svg-doc" viewBox="0 0 24 24"> <title>Document</title> <svg xmlns="http://www.w3.org/2000/svg" width="24" height="24" viewBox="0 0 24 24" fill="none" stroke="currentColor" stroke-width="2" stroke-linecap="round" stroke-linejoin="round" class="feather feather-file"> <path d="M13 2H6a2 2 0 0 0-2 2v16a2 2 0 0 0 2 2h12a2 2 0 0 0 2-2V9z"></path><polyline points="13 2 13 9 20 9"></polyline> </svg> </symbol> </svg> <div class="side-bar"> <div class="site-header"> <a href="/" class="site-title lh-tight"> <div class="site-logo"></div> </a> <a href="#" id="menu-button" class="site-button"> <svg viewBox="0 0 24 24" class="icon"><use xlink:href="#svg-menu"></use></svg> </a> </div> <nav role="navigation" aria-label="Main" id="site-nav" class="site-nav"> <ul class="nav-list"><li class="nav-list-item"><a href="/" class="nav-list-link">Home</a></li><li class="nav-list-item"><a href="#" class="nav-list-expander"><svg viewBox="0 0 24 24"><use xlink:href="#svg-arrow-right"></use></svg></a><a href="/methods/" class="nav-list-link">Methods</a><ul class="nav-list "><li class="nav-list-item "><a href="/methods/Modules.html" class="nav-list-link">PharmCAT Modules</a></li><li class="nav-list-item "><a href="/methods/determining-alleles" class="nav-list-link">Determining Alleles</a></li><li class="nav-list-item "><a href="/methods/NamedAlleleMatcher-101.html" class="nav-list-link">Named Allele Matcher 101</a></li><li class="nav-list-item "><a href="/methods/NamedAlleleMatcher-201.html" class="nav-list-link">Named Allele Matcher 201</a></li><li class="nav-list-item "><a href="/methods/gene-definition-exceptions" class="nav-list-link">Gene Definition Exceptions</a></li><li class="nav-list-item "><a href="/methods/versioning/" class="nav-list-link">Versioning</a></li></ul></li><li class="nav-list-item active"><a href="#" class="nav-list-expander"><svg viewBox="0 0 24 24"><use xlink:href="#svg-arrow-right"></use></svg></a><a href="/using/" class="nav-list-link">Using PharmCAT</a><ul class="nav-list "><li class="nav-list-item "><a href="/using/Running-PharmCAT.html" class="nav-list-link">Running PharmCAT</a></li><li class="nav-list-item "><a href="/using/VCF-Preprocessor.html" class="nav-list-link">VCF Preprocessor</a></li><li class="nav-list-item "><a href="/using/PharmCAT-in-Docker.html" class="nav-list-link">PharmCAT in Docker</a></li><li class="nav-list-item "><a href="/using/Multi-Sample-Analysis.html" class="nav-list-link">Multi-Sample Analysis</a></li><li class="nav-list-item active"><a href="/using/Calling-CYP2D6.html" class="nav-list-link active">Calling CYP2D6</a></li></ul></li><li class="nav-list-item"><a href="#" class="nav-list-expander"><svg viewBox="0 0 24 24"><use xlink:href="#svg-arrow-right"></use></svg></a><a href="/specifications/" class="nav-list-link">Specifications</a><ul class="nav-list "><li class="nav-list-item "><a href="/specifications/VCF-Requirements.html" class="nav-list-link">VCF Requirements</a></li><li class="nav-list-item "><a href="/specifications/preparing-vcf" class="nav-list-link">Preparing VCF</a></li><li class="nav-list-item "><a href="/specifications/Outside-Call-Format.html" class="nav-list-link">Outside Call Format</a></li></ul></li><li class="nav-list-item"><a href="/genes-drugs/" class="nav-list-link">Genes & Drugs</a></li><li class="nav-list-item"><a href="/examples/" class="nav-list-link">Examples</a></li><li class="nav-list-item"><a href="/faqs/" class="nav-list-link">FAQs</a></li><li class="nav-list-item"><a href="/disclaimers/" class="nav-list-link">Disclaimers and Other Information</a></li><li class="nav-list-item"><a href="/about-us/" class="nav-list-link">About Us</a></li></ul> </nav> <footer class="site-footer"> This site uses <a href="https://github.com/pmarsceill/just-the-docs">Just the Docs</a>, a documentation theme for Jekyll. </footer> </div> <div class="main" id="top"> <div id="main-header" class="main-header"> <div class="search"> <div class="search-input-wrap"> <input type="text" id="search-input" class="search-input" tabindex="0" placeholder="Search PharmCAT" aria-label="Search PharmCAT" autocomplete="off"> <label for="search-input" class="search-label"><svg viewBox="0 0 24 24" class="search-icon"><use xlink:href="#svg-search"></use></svg></label> </div> <div id="search-results" class="search-results"></div> </div> </div> <div id="main-content-wrap" class="main-content-wrap"> <nav aria-label="Breadcrumb" class="breadcrumb-nav"> <ol class="breadcrumb-nav-list"> <li class="breadcrumb-nav-list-item"><a href="/using/">Using PharmCAT</a></li> <li class="breadcrumb-nav-list-item"><span>Calling CYP2D6</span></li> </ol> </nav> <div id="main-content" class="main-content" role="main"> <h1 id="calling-cyp2d6"> <a href="#calling-cyp2d6" class="anchor-heading" aria-labelledby="calling-cyp2d6"><svg viewBox="0 0 16 16" aria-hidden="true"><use xlink:href="#svg-link"></use></svg></a> Calling CYP2D6 </h1> <p>While PharmCAT supports CYP2D6, we <strong>do NOT recommend calling CYP2D6 from VCF</strong> due to the large influence of structural variation and copy number variation (CNV) on predicted phenotype, which is beyond the scope of what can be called from SNPs in a VCF. In particular, <strong>we</strong> <strong><em>strongly discourage</em></strong> <strong>using whole-exome sequencing or low coverage whole-genome sequencing data to call CYP2D6</strong>.</p> <h2 id="why-calling-cyp2d6-from-vcf-is-a-bad-idea"> <a href="#why-calling-cyp2d6-from-vcf-is-a-bad-idea" class="anchor-heading" aria-labelledby="why-calling-cyp2d6-from-vcf-is-a-bad-idea"><svg viewBox="0 0 16 16" aria-hidden="true"><use xlink:href="#svg-link"></use></svg></a> Why calling CYP2D6 from VCF is a Bad Idea </h2> <p>For demonstration purposes, we are using <a href="https://github.com/SBIMB/StellarPGx">StellarPGx</a>, which supports calling CYP2D6 with structural variants and CNV from whole genome sequencing (WGS) CRAM/BAM files. StellarPGx has been benchmarked against the CDC's <a href="https://www.cdc.gov/labquality/get-rm/inherited-genetic-diseases-pharmacogenetics/pharmacogenetics.html">GeT-RM</a> dataset which provides ground truth sequencing data for benchmarking variant calling tools. According to the <a href="https://ascpt.onlinelibrary.wiley.com/doi/full/10.1002/cpt.2173">StellarPGx paper</a>, StellarPGx produces results with a higher concordance to GeT-RM than other available CYP2D6 callers. However, this benchmark was performed using GeT-RM's short-read 30x PCR-free WGS dataset. In addition to high-coverage WGS, GeT-RM also provides access to low-coverage WGS and whole exome sequencing (WES) data for some samples. We selected 6 random Get-RM samples and ran StellarPGx on the high-coverage WGS, low-coverage WGS, and WES and compared the results to the GeT-RM ground truth calls.</p> <div class="table-wrapper"><table> <thead> <tr> <th>Sample</th> <th>WES</th> <th>Low-Cov WGS</th> <th>30x WGS</th> <th>GeT-RM</th> </tr> </thead> <tbody> <tr> <td><a href="https://www.internationalgenome.org/data-portal/sample/HG00436">HG00436</a></td> <td>No_call</td> <td>No_call</td> <td><em>71/</em>2x2</td> <td><em>2x2/</em>71</td> </tr> <tr> <td><a href="https://www.internationalgenome.org/data-portal/sample/HG01086">HG01086</a></td> <td><em>1/</em>1x2</td> <td>No_call</td> <td><em>1/</em>31</td> <td><em>1/</em>31</td> </tr> <tr> <td><a href="https://www.internationalgenome.org/data-portal/sample/HG01190">HG01190</a></td> <td><em>1/</em>1</td> <td><em>5/</em>1</td> <td><em>5/</em>68+*4</td> <td><em>68+</em>4/*5</td> </tr> <tr> <td><a href="https://www.internationalgenome.org/data-portal/sample/NA07048">NA07048</a></td> <td>No_call</td> <td><em>1/</em>1</td> <td><em>139/</em>4</td> <td><em>1/</em>4</td> </tr> <tr> <td><a href="https://www.internationalgenome.org/data-portal/sample/NA18545">NA18545</a></td> <td><em>34/</em>34x7</td> <td><em>1/</em>1x5</td> <td><em>36+</em>10/<em>36+</em>10</td> <td><em>5/</em>36x2+*10x2</td> </tr> <tr> <td><a href="https://www.internationalgenome.org/data-portal/sample/NA21105">NA21105</a></td> <td>No_call</td> <td><em>34/</em>34</td> <td><em>111/</em>3</td> <td><em>3/</em>111</td> </tr> </tbody> </table></div> <p>There is drastic variation in the calls produced by the three sequencing technologies. In many cases, the WES or low-coverage WGS input resulted in a "no call" result. Occasionally they produced improbably high copy numbers. They also produced star-allele calls which varied greatly from the GeT-RM calls. The 30x WGS produced identical or equivalent results most of the time, and when it deviated the differences were relatively minor. In conclusion, <strong>we do not recommend the use WES or low-coverage WGS for calling CYP2D6 for both research and clinical purposes</strong>.</p> <h2 id="calling-cyp2d6-anyway"> <a href="#calling-cyp2d6-anyway" class="anchor-heading" aria-labelledby="calling-cyp2d6-anyway"><svg viewBox="0 0 16 16" aria-hidden="true"><use xlink:href="#svg-link"></use></svg></a> Calling CYP2D6 Anyway </h2> <p>So now that you know the pitfalls, if you still want to call CYP2D6, you have 2 options:</p> <ul> <li>to <a href="#using-outside-calls">use a different tool to call CYP2D6</a></li> <li>to run PharmCAT in <a href="#calling-cyp2d6-with-pharmcats-research-mode">research mode</a></li> </ul> <h3 id="using-outside-calls"> <a href="#using-outside-calls" class="anchor-heading" aria-labelledby="using-outside-calls"><svg viewBox="0 0 16 16" aria-hidden="true"><use xlink:href="#svg-link"></use></svg></a> Using Outside Calls </h3> <p>PharmCAT supports pulling in results from other tools using what we call <a href="/specifications/Outside-Call-Format">"outside calls"</a>. This allows you to select your preferred CYP2D6 caller and include CYP2D6 results in a PharmCAT report.</p> <p>To include external calls you need to use the <code class="language-plaintext highlighter-rouge">-po</code> flag in PharmCAT to specify the external calls file. For example:</p><pre><code class="language-commandline"># java -jar pharmcat.jar -vcf patient_001.vcf -po patient_001_cyp2d6.txt
</code></pre><p>where <code class="language-plaintext highlighter-rouge">patient_001_cyp2d6.txt</code> takes the format of:</p> <div class="language-text highlighter-rouge"><div class="highlight"><pre class="highlight"><code># CYP2D6	*1x2/*2+*6
</code></pre></div></div> <ul> <li>See <a href="/using/Running-PharmCAT#phenotyper">Running PharmCAT</a> for details on the <code class="language-plaintext highlighter-rouge">-po</code> flag</li> <li>See <a href="/specifications/Outside-Call-Format">Outside Call Format</a> for details on the outside call file</li> </ul> <h4 id="formatting-stellarpgx-output-for-pharmcat"> <a href="#formatting-stellarpgx-output-for-pharmcat" class="anchor-heading" aria-labelledby="formatting-stellarpgx-output-for-pharmcat"><svg viewBox="0 0 16 16" aria-hidden="true"><use xlink:href="#svg-link"></use></svg></a> Formatting StellarPGx output for PharmCAT </h4> <p>As mentioned above, if you have whole genome sequencing (WGS) CRAM/BAM files, your best option is a tool like StellarPGx. Please note we are not affiliated with the StellarPGx team and offer no guarantees about its performance. Any questions or concerns on StellarPGx should be directed to the StellarPGx maintainers <a href="twesigomwedavid@gmail.com">twesigomwedavid@gmail.com</a>. Usage instructions for StellarPGx can be found on their <a href="https://github.com/SBIMB/StellarPGx">github</a>.</p> <p>While this tutorial is StellarPGx specific, it should illustrate how to integrate other external callers with PharmCAT.</p> <p>After running StellarPGx with CYP2D6 as the target gene, it should produce a <code class="language-plaintext highlighter-rouge">&lt;run_name&gt;_summary.txt</code> file which looks like:</p> <div class="language-text highlighter-rouge"><div class="highlight"><pre class="highlight"><code>HG00436	*71/*2x2
HG01086	[*1/*31]	Possible novel allele or suballele present: interpret with caution; experimental validation and expert review through PharmVar is recommended
HG01190	*5/*68+*4
NA07048	*139/*4
NA18545	*36+*10/*36+*10
NA21105	*111/*3
</code></pre></div></div> <p>Column 1 is the sample name and column 2 is the CYP2D6 call. In order to convert this into a PharmCAT-readable format you must split this into a separate file for each sample.</p> <p>The following is a simple python script that parses StellarPGx's <code class="language-plaintext highlighter-rouge">summary.txt</code> file and outputs PharmCAT-ready external calls files for use with the <code class="language-plaintext highlighter-rouge">-a</code> flag.</p> <div class="language-python highlighter-rouge"><div class="highlight"><pre class="highlight"><code><span class="kn">import</span> <span class="nn">sys</span>
<span class="kn">import</span> <span class="nn">os</span>

<span class="n">input_path</span> <span class="o">=</span> <span class="n">sys</span><span class="p">.</span><span class="n">argv</span><span class="p">[</span><span class="mi">1</span><span class="p">]</span> <span class="c1"># summary.txt file from StellarPGx
</span><span class="n">output_dir</span> <span class="o">=</span> <span class="n">sys</span><span class="p">.</span><span class="n">argv</span><span class="p">[</span><span class="mi">2</span><span class="p">]</span> <span class="c1"># output directory
</span>
<span class="k">with</span> <span class="nb">open</span><span class="p">(</span><span class="n">input_path</span><span class="p">,</span> <span class="s">"r"</span><span class="p">)</span> <span class="k">as</span> <span class="n">infile</span><span class="p">:</span>
    <span class="n">entries</span> <span class="o">=</span> <span class="n">infile</span><span class="p">.</span><span class="n">readlines</span><span class="p">()</span> <span class="c1"># read all the StellarPGx file lines into a list
</span>
<span class="k">for</span> <span class="n">entry</span> <span class="ow">in</span> <span class="n">entries</span><span class="p">:</span> <span class="c1"># for each sample in the StellarPGx file
</span>    <span class="n">split</span> <span class="o">=</span> <span class="n">entry</span><span class="p">.</span><span class="n">split</span><span class="p">(</span><span class="s">"</span><span class="se">\t</span><span class="s">"</span><span class="p">)</span> <span class="c1"># split the lines into columns
</span>    <span class="n">sample</span> <span class="o">=</span> <span class="n">split</span><span class="p">[</span><span class="mi">0</span><span class="p">]</span> <span class="c1"># extract the sample name
</span>    <span class="n">call</span> <span class="o">=</span> <span class="n">split</span><span class="p">[</span><span class="mi">1</span><span class="p">].</span><span class="n">strip</span><span class="p">(</span><span class="s">"[] </span><span class="se">\n</span><span class="s">"</span><span class="p">)</span> <span class="c1"># extract the sample call and strip excess spaces and brackets
</span>
    <span class="k">if</span> <span class="ow">not</span> <span class="n">os</span><span class="p">.</span><span class="n">path</span><span class="p">.</span><span class="n">exists</span><span class="p">(</span><span class="n">output_dir</span><span class="p">):</span> <span class="c1"># create the output directory if it doesn't exist yet
</span>        <span class="n">os</span><span class="p">.</span><span class="n">makedir</span><span class="p">(</span><span class="n">output_dir</span><span class="p">)</span>

    <span class="k">with</span> <span class="nb">open</span><span class="p">(</span><span class="s">"%s/%s_cyp2d6.txt"</span> <span class="o">%</span> <span class="p">(</span><span class="n">output_dir</span><span class="p">,</span><span class="n">sample</span><span class="p">),</span> <span class="s">"w"</span><span class="p">)</span> <span class="k">as</span> <span class="n">outfile</span><span class="p">:</span>
        <span class="c1"># write the sample to a text file in the specified output directory
</span>        <span class="n">outfile</span><span class="p">.</span><span class="n">write</span><span class="p">(</span><span class="s">"CYP2D6</span><span class="se">\t</span><span class="s">%s"</span> <span class="o">%</span> <span class="n">call</span><span class="p">)</span>
</code></pre></div></div> <p>Usage example:</p> <div class="language-console highlighter-rouge"><div class="highlight"><pre class="highlight"><code><span class="gp">#</span><span class="w"> </span>python3 stellarPGx_to_PharmCAT.py summary.txt pharmcat_inputs
</code></pre></div></div> <p>Where <code class="language-plaintext highlighter-rouge">summary.txt</code> is the StellarPGx output and <code class="language-plaintext highlighter-rouge">pharmcat_inputs</code> is the directory where I want my call files to be placed. If I ran the script on the summary file example from above, calling <code class="language-plaintext highlighter-rouge">ls pharmcat_inputs</code> to list the directory contents should show:</p> <div class="language-console highlighter-rouge"><div class="highlight"><pre class="highlight"><code><span class="gp">#</span><span class="w"> </span><span class="nb">ls </span>pharmcat_inputs
<span class="go">HG00436_cyp2d6.txt  HG01086_cyp2d6.txt  HG01190_cyp2d6.txt  NA07048_cyp2d6.txt  NA18545_cyp2d6.txt  NA21105_cyp2d6.txt
</span></code></pre></div></div> <p>and <code class="language-plaintext highlighter-rouge">HG00436_cyp2d6.txt</code> would look like</p> <div class="language-text highlighter-rouge"><div class="highlight"><pre class="highlight"><code>CYP2D6  *71/*2x2
</code></pre></div></div> <p>We urge you to manually verify that the calls in the PharmCAT-ready files correspond to the StellarPGx output.</p> <h3 id="calling-cyp2d6-with-pharmcats-research-mode"> <a href="#calling-cyp2d6-with-pharmcats-research-mode" class="anchor-heading" aria-labelledby="calling-cyp2d6-with-pharmcats-research-mode"><svg viewBox="0 0 16 16" aria-hidden="true"><use xlink:href="#svg-link"></use></svg></a> Calling CYP2D6 with PharmCAT's research mode </h3> <p>Once again, we <strong>do NOT recommend calling CYP2D6 from VCF</strong>!</p> <p>PharmCAT can, however, call CYP2D6 if you are willing to accept all the caveats. This functionality is gated behind the <code class="language-plaintext highlighter-rouge">-research</code> flag.</p> <p>To get PharmCAT to call CYP2D6, use <code class="language-plaintext highlighter-rouge">-research cyp2d6</code></p> <div class="language-console highlighter-rouge"><div class="highlight"><pre class="highlight"><code><span class="gp">#</span><span class="w"> </span>java <span class="nt">-jar</span> pharmcat.jar <span class="nt">-vcf</span> patient_001.vcf <span class="nt">-research</span> cyp2d6
</code></pre></div></div> <hr> <footer> <div> PharmCAT is managed at Stanford University & University of Pennsylvania (NHGRI U24HG010862). </div> </footer> </div> </div> <div class="search-overlay"></div> </div> </body> </html>
