<!DOCTYPE html> <html lang="en-US"> <head> <meta charset="UTF-8"> <meta http-equiv="X-UA-Compatible" content="IE=Edge"> <title>Multi-Sample Analysis - PharmCAT</title> <link rel="shortcut icon" href="/favicon.ico" type="image/x-icon"> <link rel="stylesheet" href="/assets/css/just-the-docs-default.css"> <script async src="https://www.googletagmanager.com/gtag/js?id=G-7ZP1HHBWTE"></script> <script> window.dataLayer = window.dataLayer || []; function gtag(){dataLayer.push(arguments);} gtag('js', new Date()); gtag('config', 'G-7ZP1HHBWTE', { 'anonymize_ip': true }); </script> <script type="text/javascript" src="/assets/js/vendor/lunr.min.js"></script> <script type="text/javascript" src="/assets/js/just-the-docs.js"></script> <meta name="viewport" content="width=device-width, initial-scale=1"> <!-- Begin Jekyll SEO tag v2.8.0 --> <title>Multi-Sample Analysis | PharmCAT</title> <meta name="generator" content="Jekyll v4.2.2" /> <meta property="og:title" content="Multi-Sample Analysis" /> <meta property="og:locale" content="en_US" /> <meta name="description" content="The Pharmacogenomics Clinical Annotation Tool (PharmCAT) is an open-source tool that takes an individual’s genetic data and determines their respective drug-related genotypes and phenotypes. PharmCAT produces a personalized report with genotype-specific expert-reviewed drug prescribing recommendations." /> <meta property="og:description" content="The Pharmacogenomics Clinical Annotation Tool (PharmCAT) is an open-source tool that takes an individual’s genetic data and determines their respective drug-related genotypes and phenotypes. PharmCAT produces a personalized report with genotype-specific expert-reviewed drug prescribing recommendations." /> <meta property="og:site_name" content="PharmCAT" /> <meta property="og:type" content="website" /> <meta name="twitter:card" content="summary" /> <meta property="twitter:title" content="Multi-Sample Analysis" /> <script type="application/ld+json"> {"@context":"https://schema.org","@type":"WebPage","description":"The Pharmacogenomics Clinical Annotation Tool (PharmCAT) is an open-source tool that takes an individual’s genetic data and determines their respective drug-related genotypes and phenotypes. PharmCAT produces a personalized report with genotype-specific expert-reviewed drug prescribing recommendations.","headline":"Multi-Sample Analysis","publisher":{"@type":"Organization","logo":{"@type":"ImageObject","url":"/images/pharmcat_header.png"}},"url":"/using/Multi-Sample-Analysis.html"}</script> <!-- End Jekyll SEO tag --> <script async defer data-website-id="714c506c-5f1e-4ea8-af79-fa80c02911c6" src="https://metrics.cpicpgx.org/j.js"></script> <style> h4 { font-size: 14px !important; font-weight: bold; text-transform: none; } h4 code { font-size: 0.9em; font-weight: bold; } h5 code { font-size: .9em; } .main-content ul, .main-content ol { padding-left: 2em; } .main-content dl { grid-template-columns: 16em 1fr; } .main-content dt::after { content: unset; } .altArg { font-size: .8em; font-weight: normal; } footer { position: relative; top: -1.5em; font-size: smaller; } </style> </head> <body> <svg xmlns="http://www.w3.org/2000/svg" style="display: none;"> <symbol id="svg-link" viewBox="0 0 24 24"> <title>Link</title> <svg xmlns="http://www.w3.org/2000/svg" width="24" height="24" viewBox="0 0 24 24" fill="none" stroke="currentColor" stroke-width="2" stroke-linecap="round" stroke-linejoin="round" class="feather feather-link"> <path d="M10 13a5 5 0 0 0 7.54.54l3-3a5 5 0 0 0-7.07-7.07l-1.72 1.71"></path><path d="M14 11a5 5 0 0 0-7.54-.54l-3 3a5 5 0 0 0 7.07 7.07l1.71-1.71"></path> </svg> </symbol> <symbol id="svg-search" viewBox="0 0 24 24"> <title>Search</title> <svg xmlns="http://www.w3.org/2000/svg" width="24" height="24" viewBox="0 0 24 24" fill="none" stroke="currentColor" stroke-width="2" stroke-linecap="round" stroke-linejoin="round" class="feather feather-search"> <circle cx="11" cy="11" r="8"></circle><line x1="21" y1="21" x2="16.65" y2="16.65"></line> </svg> </symbol> <symbol id="svg-menu" viewBox="0 0 24 24"> <title>Menu</title> <svg xmlns="http://www.w3.org/2000/svg" width="24" height="24" viewBox="0 0 24 24" fill="none" stroke="currentColor" stroke-width="2" stroke-linecap="round" stroke-linejoin="round" class="feather feather-menu"> <line x1="3" y1="12" x2="21" y2="12"></line><line x1="3" y1="6" x2="21" y2="6"></line><line x1="3" y1="18" x2="21" y2="18"></line> </svg> </symbol> <symbol id="svg-arrow-right" viewBox="0 0 24 24"> <title>Expand</title> <svg xmlns="http://www.w3.org/2000/svg" width="24" height="24" viewBox="0 0 24 24" fill="none" stroke="currentColor" stroke-width="2" stroke-linecap="round" stroke-linejoin="round" class="feather feather-chevron-right"> <polyline points="9 18 15 12 9 6"></polyline> </svg> </symbol> <symbol id="svg-doc" viewBox="0 0 24 24"> <title>Document</title> <svg xmlns="http://www.w3.org/2000/svg" width="24" height="24" viewBox="0 0 24 24" fill="none" stroke="currentColor" stroke-width="2" stroke-linecap="round" stroke-linejoin="round" class="feather feather-file"> <path d="M13 2H6a2 2 0 0 0-2 2v16a2 2 0 0 0 2 2h12a2 2 0 0 0 2-2V9z"></path><polyline points="13 2 13 9 20 9"></polyline> </svg> </symbol> </svg> <div class="side-bar"> <div class="site-header"> <a href="/" class="site-title lh-tight"> <div class="site-logo"></div> </a> <a href="#" id="menu-button" class="site-button"> <svg viewBox="0 0 24 24" class="icon"><use xlink:href="#svg-menu"></use></svg> </a> </div> <nav role="navigation" aria-label="Main" id="site-nav" class="site-nav"> <ul class="nav-list"><li class="nav-list-item"><a href="/" class="nav-list-link">Home</a></li><li class="nav-list-item"><a href="#" class="nav-list-expander"><svg viewBox="0 0 24 24"><use xlink:href="#svg-arrow-right"></use></svg></a><a href="/methods/" class="nav-list-link">Methods</a><ul class="nav-list "><li class="nav-list-item "><a href="/methods/Modules.html" class="nav-list-link">PharmCAT Modules</a></li><li class="nav-list-item "><a href="/methods/determining-alleles" class="nav-list-link">Determining Alleles</a></li><li class="nav-list-item "><a href="/methods/NamedAlleleMatcher-101.html" class="nav-list-link">Named Allele Matcher 101</a></li><li class="nav-list-item "><a href="/methods/NamedAlleleMatcher-201.html" class="nav-list-link">Named Allele Matcher 201</a></li><li class="nav-list-item "><a href="/methods/gene-definition-exceptions" class="nav-list-link">Gene Definition Exceptions</a></li><li class="nav-list-item "><a href="/methods/versioning/" class="nav-list-link">Versioning</a></li></ul></li><li class="nav-list-item active"><a href="#" class="nav-list-expander"><svg viewBox="0 0 24 24"><use xlink:href="#svg-arrow-right"></use></svg></a><a href="/using/" class="nav-list-link">Using PharmCAT</a><ul class="nav-list "><li class="nav-list-item "><a href="/using/Running-PharmCAT.html" class="nav-list-link">Running PharmCAT</a></li><li class="nav-list-item "><a href="/using/VCF-Preprocessor.html" class="nav-list-link">VCF Preprocessor</a></li><li class="nav-list-item "><a href="/using/PharmCAT-in-Docker.html" class="nav-list-link">PharmCAT in Docker</a></li><li class="nav-list-item active"><a href="/using/Multi-Sample-Analysis.html" class="nav-list-link active">Multi-Sample Analysis</a></li><li class="nav-list-item "><a href="/using/Calling-CYP2D6.html" class="nav-list-link">Calling CYP2D6</a></li></ul></li><li class="nav-list-item"><a href="#" class="nav-list-expander"><svg viewBox="0 0 24 24"><use xlink:href="#svg-arrow-right"></use></svg></a><a href="/specifications/" class="nav-list-link">Specifications</a><ul class="nav-list "><li class="nav-list-item "><a href="/specifications/VCF-Requirements.html" class="nav-list-link">VCF Requirements</a></li><li class="nav-list-item "><a href="/specifications/preparing-vcf" class="nav-list-link">Preparing VCF</a></li><li class="nav-list-item "><a href="/specifications/Outside-Call-Format.html" class="nav-list-link">Outside Call Format</a></li></ul></li><li class="nav-list-item"><a href="/genes-drugs/" class="nav-list-link">Genes & Drugs</a></li><li class="nav-list-item"><a href="/examples/" class="nav-list-link">Examples</a></li><li class="nav-list-item"><a href="/faqs/" class="nav-list-link">FAQs</a></li><li class="nav-list-item"><a href="/disclaimers/" class="nav-list-link">Disclaimers and Other Information</a></li><li class="nav-list-item"><a href="/about-us/" class="nav-list-link">About Us</a></li></ul> </nav> <footer class="site-footer"> This site uses <a href="https://github.com/pmarsceill/just-the-docs">Just the Docs</a>, a documentation theme for Jekyll. </footer> </div> <div class="main" id="top"> <div id="main-header" class="main-header"> <div class="search"> <div class="search-input-wrap"> <input type="text" id="search-input" class="search-input" tabindex="0" placeholder="Search PharmCAT" aria-label="Search PharmCAT" autocomplete="off"> <label for="search-input" class="search-label"><svg viewBox="0 0 24 24" class="search-icon"><use xlink:href="#svg-search"></use></svg></label> </div> <div id="search-results" class="search-results"></div> </div> </div> <div id="main-content-wrap" class="main-content-wrap"> <nav aria-label="Breadcrumb" class="breadcrumb-nav"> <ol class="breadcrumb-nav-list"> <li class="breadcrumb-nav-list-item"><a href="/using/">Using PharmCAT</a></li> <li class="breadcrumb-nav-list-item"><span>Multi-Sample Analysis</span></li> </ol> </nav> <div id="main-content" class="main-content" role="main"> <h1 class="no_toc" id="multi-sample-analysis"> <a href="#multi-sample-analysis" class="anchor-heading" aria-labelledby="multi-sample-analysis"><svg viewBox="0 0 16 16" aria-hidden="true"><use xlink:href="#svg-link"></use></svg></a> Multi-Sample Analysis </h1> <p>As of March 2022, PharmCAT only takes a single-sample VCF. If a multi-sample VCF is provided, only the first sample will be annotated. However, PharmCAT can generate a PGx report in a matter of seconds for a preprocessed VCF. The fast runtime of PharmCAT allows you to batch-annotate VCFs, with a little help from a bit of scripting, to scale from a few dozen samples to a biobank-scale cohorts in an efficient manner.</p> <p>This documentation shares concrete examples of how to use PharmCAT to batch process multiple samples in a High-Performance Computing (HPC) environment.</p> <p>We expect readers to be familiar with using both PharmCAT's <a href="/using/VCF-Preprocessor">VCF preprocessor</a> and the core <a href="/using/Running-PharmCAT">PharmCAT tool</a>. In addition to the documentation, we also have a <a href="https://github.com/PharmGKB/PharmCAT-tutorial">tutorial</a> available that walks you through working with real genetic data sets.</p><hr /> <h2 class="no_toc text-delta" id="table-of-contents"> <a href="#table-of-contents" class="anchor-heading" aria-labelledby="table-of-contents"><svg viewBox="0 0 16 16" aria-hidden="true"><use xlink:href="#svg-link"></use></svg></a> Table of contents </h2> <ol id="markdown-toc"> <li><a href="#example-using-the-stanford-sherlock-hpc" id="markdown-toc-example-using-the-stanford-sherlock-hpc">Example using the Stanford Sherlock HPC</a> <ol> <li><a href="#preprocessing-the-data" id="markdown-toc-preprocessing-the-data">Preprocessing the data</a> <ol> <li><a href="#case-1---single-sample-vcfs" id="markdown-toc-case-1---single-sample-vcfs">Case 1 - single-sample VCFs</a></li> <li><a href="#case-2---multi-sample-vcf" id="markdown-toc-case-2---multi-sample-vcf">Case 2 - multi-sample VCF</a></li> <li><a href="#case-3---multi-sample-vcf-divided-by-chromosome-or-into-consecutive-genetic-blocks" id="markdown-toc-case-3---multi-sample-vcf-divided-by-chromosome-or-into-consecutive-genetic-blocks">Case 3 - multi-sample VCF divided by chromosome or into consecutive genetic blocks</a></li> </ol> </li> <li><a href="#running-pharmcat" id="markdown-toc-running-pharmcat">Running PharmCAT</a> <ol> <li><a href="#batch-outside-calls" id="markdown-toc-batch-outside-calls">Batch outside calls</a></li> </ol> </li> <li><a href="#running-individual-components" id="markdown-toc-running-individual-components">Running individual components</a> <ol> <li><a href="#named-allele-matcher" id="markdown-toc-named-allele-matcher">Named Allele Matcher</a></li> <li><a href="#phenotyper---use-the-named-allele-matcher-json-data" id="markdown-toc-phenotyper---use-the-named-allele-matcher-json-data">Phenotyper - use the Named Allele Matcher JSON data</a></li> <li><a href="#reporter" id="markdown-toc-reporter">Reporter</a></li> </ol> </li> <li><a href="#extracting-pharmcat-json-content-into-tsv" id="markdown-toc-extracting-pharmcat-json-content-into-tsv">Extracting PharmCAT JSON content into TSV</a> <ol> <li><a href="#extracting-the-pharmcat-named-allele-matcher-json-data-into-a-tsv-file" id="markdown-toc-extracting-the-pharmcat-named-allele-matcher-json-data-into-a-tsv-file">Extracting the PharmCAT Named Allele Matcher JSON data into a TSV file</a></li> <li><a href="#extracting-the-pharmcat-phenotyper-json-data-into-a-tsv-file" id="markdown-toc-extracting-the-pharmcat-phenotyper-json-data-into-a-tsv-file">Extracting the PharmCAT Phenotyper JSON data into a TSV file</a></li> </ol> </li> </ol> </li> <li><a href="#example-with-the-penn-medicine-biobank" id="markdown-toc-example-with-the-penn-medicine-biobank">Example with the Penn Medicine Biobank</a> <ol> <li><a href="#preprocessing-the-data-1" id="markdown-toc-preprocessing-the-data-1">Preprocessing the data</a></li> <li><a href="#running-pharmcat-1" id="markdown-toc-running-pharmcat-1">Running PharmCAT</a></li> <li><a href="#reading-the-pharmcat-calls-into-python" id="markdown-toc-reading-the-pharmcat-calls-into-python">Reading the PharmCAT calls into Python</a></li> </ol> </li> <li><a href="#reference" id="markdown-toc-reference">Reference</a></li> </ol><hr /> <h2 id="example-using-the-stanford-sherlock-hpc"> <a href="#example-using-the-stanford-sherlock-hpc" class="anchor-heading" aria-labelledby="example-using-the-stanford-sherlock-hpc"><svg viewBox="0 0 16 16" aria-hidden="true"><use xlink:href="#svg-link"></use></svg></a> Example using the Stanford Sherlock HPC </h2> <p>This part uses the VCFs on multiple GeT-RM samples<a href="#reference">1</a> generated from the 30x high-coverage WGS sequencing performed by the New York Genome Center (NYGC)<a href="#reference">2</a>. The job was run on the Stanford Sherlock cluster, a High-Performance Computing (HPC) cluster which uses Slurm as an open-source resource manager and job scheduler. The scripts to run PharmCAT on the Stanford Sherlock cluster is written in shell programming language and can be easily adapted to another HPC environment supported by a different job scheduler.</p> <h3 id="preprocessing-the-data"> <a href="#preprocessing-the-data" class="anchor-heading" aria-labelledby="preprocessing-the-data"><svg viewBox="0 0 16 16" aria-hidden="true"><use xlink:href="#svg-link"></use></svg></a> Preprocessing the data </h3> <p>This section provides examples for working with different types of input VCFs. For the full script, see <a href="https://github.com/PharmGKB/PharmCAT-tutorial/blob/main/src/">PharmCAT-tutorial/src/02_VCF_preprocessing.sh</a>.</p> <h4 id="case-1---single-sample-vcfs"> <a href="#case-1---single-sample-vcfs" class="anchor-heading" aria-labelledby="case-1---single-sample-vcfs"><svg viewBox="0 0 16 16" aria-hidden="true"><use xlink:href="#svg-link"></use></svg></a> Case 1 - single-sample VCFs </h4> <p>If your genetic data is already stored in single-sample VCFs, you are one step closer to running the PharmCAT. We recommend the users to still run the PharmCAT VCF preprocessor to ensure the appropriate variant representation format, which can be achieved by the following command.</p> <p>On the command line:</p> <div class="language-console highlighter-rouge"><div class="highlight"><pre class="highlight"><code><span class="gp">$</span><span class="w"> </span>python3 PharmCAT_VCF_Preprocess.py <span class="nt">--input_vcf</span> &lt;single_sample_vcf&gt;
</code></pre></div></div> <p>For example:</p> <div class="language-console highlighter-rouge"><div class="highlight"><pre class="highlight"><code><span class="gp">$</span><span class="w"> </span><span class="nb">cd </span>PharmCAT-tutorial/
<span class="gp">$</span><span class="w"> </span><span class="nb">cat </span>data/single_sample_vcf_list.txt
<span class="go">data/PharmCAT_tutorial_get-rm_wgs_30x_grch38.NA18526.vcf.gz
data/PharmCAT_tutorial_get-rm_wgs_30x_grch38.NA18565.vcf.gz
data/PharmCAT_tutorial_get-rm_wgs_30x_grch38.NA18861.vcf.gz
</span></code></pre></div></div> <p>Sample script:</p> <div class="language-shell highlighter-rouge"><div class="highlight"><pre class="highlight"><code><span class="c"># run the preprocessor iteratively across single-sample VCFs</span>
<span class="k">for </span>SINGLE_VCF <span class="k">in</span> <span class="si">$(</span><span class="nb">cat </span>data/single_sample_vcf_list.txt<span class="si">)</span>
<span class="k">do</span>
  <span class="c"># run the PharmCAT VCF preprocessor for a single-sample VCF</span>
  python3 PharmCAT_VCF_Preprocess.py <span class="nt">--input_vcf</span> <span class="s2">"</span><span class="nv">$SINGLE_VCF</span><span class="s2">"</span>
<span class="k">done</span>
</code></pre></div></div> <h4 id="case-2---multi-sample-vcf"> <a href="#case-2---multi-sample-vcf" class="anchor-heading" aria-labelledby="case-2---multi-sample-vcf"><svg viewBox="0 0 16 16" aria-hidden="true"><use xlink:href="#svg-link"></use></svg></a> Case 2 - multi-sample VCF </h4> <p>Population- or biobank-scale VCFs most likely come in multi-sample format. The PharmCAT VCF preprocessor is designed to help the users with this case and produce multiple single-sample VCFs that PharmCAT requires. The simplest command to preprocess a multi-sample VCF is as follows.</p> <p>On the command line:</p> <div class="language-console highlighter-rouge"><div class="highlight"><pre class="highlight"><code><span class="gp">$</span><span class="w"> </span>python3 PharmCAT_VCF_Preprocess.py <span class="nt">--input_vcf</span> &lt;multi_sample_vcf&gt;
</code></pre></div></div> <p>Sample script:</p> <div class="language-shell highlighter-rouge"><div class="highlight"><pre class="highlight"><code><span class="nb">cd </span>PharmCAT-tutorial/
python3 PharmCAT_VCF_Preprocess.py <span class="nt">--input_vcf</span> data/PharmCAT_tutorial_get-rm_wgs_30x_grch38.vcf.gz
</code></pre></div></div> <h4 id="case-3---multi-sample-vcf-divided-by-chromosome-or-into-consecutive-genetic-blocks"> <a href="#case-3---multi-sample-vcf-divided-by-chromosome-or-into-consecutive-genetic-blocks" class="anchor-heading" aria-labelledby="case-3---multi-sample-vcf-divided-by-chromosome-or-into-consecutive-genetic-blocks"><svg viewBox="0 0 16 16" aria-hidden="true"><use xlink:href="#svg-link"></use></svg></a> Case 3 - multi-sample VCF divided by chromosome or into consecutive genetic blocks </h4> <p>As sometimes seen with large-scale genetic studies, the genetic data may be divided into multiple by-chromosome VCFs or VCFs with consecutive genetic blocks. The PharmCAT VCF preprocessor can manage this type of genetic data sets by taking a list of VCFs as the input.</p> <p>On the command line:</p> <div class="language-console highlighter-rouge"><div class="highlight"><pre class="highlight"><code><span class="gp">$</span><span class="w"> </span>python3 PharmCAT_VCF_Preprocess.py <span class="nt">--input_list</span> &lt;list_of_input_vcf&gt;
</code></pre></div></div> <p>Sample script:</p> <div class="language-shell highlighter-rouge"><div class="highlight"><pre class="highlight"><code><span class="nb">cd </span>PharmCAT-tutorial/
<span class="c"># run the PharmCAT VCF preprocessor for multiple VCFs with non-overlapping genetic regions of the same cohort</span>
python3 PharmCAT_VCF_Preprocess.py <span class="nt">--input_list</span> data/input_vcf_list.txt
</code></pre></div></div> <h3 id="running-pharmcat"> <a href="#running-pharmcat" class="anchor-heading" aria-labelledby="running-pharmcat"><svg viewBox="0 0 16 16" aria-hidden="true"><use xlink:href="#svg-link"></use></svg></a> Running PharmCAT </h3> <p>Assuming the users have run the PharmCAT VCF preprocessor to generate multiple single-sample VCFs which is named such as <code class="language-plaintext highlighter-rouge">pharmcat_ready_vcf.&lt;sample_id&gt;.vcf</code>. Use the following command to batch annotate multiple VCFs using PharmCAT. A full example can be found at <a href="https://github.com/PharmGKB/PharmCAT-tutorial/blob/main/src/03_PharmCAT.sh">PharmCAT-tutorial/src/03_PharmCAT.sh</a>.</p> <p>On the command line:</p> <div class="language-console highlighter-rouge"><div class="highlight"><pre class="highlight"><code><span class="gp">$</span><span class="w"> </span>java <span class="nt">-jar</span> &lt;path_to_the_latest_pharmcat_jar&gt; <span class="nt">-vcf</span> &lt;single_sample_vcf&gt;
</code></pre></div></div> <p>Sample script:</p> <div class="language-shell highlighter-rouge"><div class="highlight"><pre class="highlight"><code><span class="nb">cd </span>PharmCAT-tutorial/
<span class="c"># running multiple samples</span>
<span class="k">for </span>SINGLE_SAMPLE <span class="k">in</span> <span class="si">$(</span><span class="nb">cat </span>data/test_get-rm_samples.txt<span class="si">)</span>
<span class="k">do</span>
  <span class="c"># always use the latest PharmCAT</span>
  java <span class="nt">-jar</span> pharmcat.jar <span class="nt">-vcf</span> results/pharmcat_ready/pharmcat_ready.<span class="s2">"</span><span class="nv">$SINGLE_SAMPLE</span><span class="s2">"</span>.vcf
<span class="k">done</span>
</code></pre></div></div> <p>The output is a set of PGx reports in HTML format named as <code class="language-plaintext highlighter-rouge">pharmcat.&lt;sample_id&gt;.report.html</code>.</p> <h4 id="batch-outside-calls"> <a href="#batch-outside-calls" class="anchor-heading" aria-labelledby="batch-outside-calls"><svg viewBox="0 0 16 16" aria-hidden="true"><use xlink:href="#svg-link"></use></svg></a> Batch outside calls </h4> <p>To incorporate outside PGx calls with PharmCAT for multiple samples, the users have to prepare the outside PGx calls of each sample in separate files and supply the individual file as outside calls to the PharmCAT.</p> <h3 id="running-individual-components"> <a href="#running-individual-components" class="anchor-heading" aria-labelledby="running-individual-components"><svg viewBox="0 0 16 16" aria-hidden="true"><use xlink:href="#svg-link"></use></svg></a> Running individual components </h3> <p>You can run the individual modules of PharmCAT on multiple samples in a similar manner to how you run the whole PharmCAT pipeline. This is useful if you are interested in understanding population PGx and specifically obtaining PGx frequencies (named alleles, diplotypes, or metabolizer phenotypes) in your cohort. To do so, you can run the following commands against your data. More details and a tutorial can be found at <a href="https://github.com/PharmGKB/PharmCAT-tutorial/blob/main/src/03_PharmCAT.sh">PharmCAT-tutorial/src/03_PharmCAT.sh</a>.</p> <h4 id="named-allele-matcher"> <a href="#named-allele-matcher" class="anchor-heading" aria-labelledby="named-allele-matcher"><svg viewBox="0 0 16 16" aria-hidden="true"><use xlink:href="#svg-link"></use></svg></a> Named Allele Matcher </h4> <p>On the command line:</p> <div class="language-console highlighter-rouge"><div class="highlight"><pre class="highlight"><code><span class="gp">$</span><span class="w"> </span>java <span class="nt">-jar</span> &lt;path_to_the_latest_pharmcat_jar&gt; <span class="nt">-matcher</span> <span class="nt">-vcf</span> &lt;sample.vcf&gt;
</code></pre></div></div> <p>Sample script:</p> <div class="language-shell highlighter-rouge"><div class="highlight"><pre class="highlight"><code><span class="nb">cd </span>PharmCAT-tutorial/
<span class="k">for </span>SINGLE_SAMPLE <span class="k">in</span> <span class="s2">"</span><span class="si">$(</span><span class="nb">cat </span>data/test_get-rm_samples.txt<span class="si">)</span><span class="s2">"</span>
<span class="k">do
  </span>java <span class="nt">-jar</span> pharmcat.jar <span class="nt">-matcher</span> <span class="nt">-vcf</span> results/pharmcat_ready/pharmcat_ready_vcf.<span class="s2">"</span><span class="nv">$SINGLE_SAMPLE</span><span class="s2">"</span>.vcf
<span class="k">done</span>
</code></pre></div></div> <h4 id="phenotyper---use-the-named-allele-matcher-json-data"> <a href="#phenotyper---use-the-named-allele-matcher-json-data" class="anchor-heading" aria-labelledby="phenotyper---use-the-named-allele-matcher-json-data"><svg viewBox="0 0 16 16" aria-hidden="true"><use xlink:href="#svg-link"></use></svg></a> Phenotyper - use the Named Allele Matcher JSON data </h4> <p>On the command line:</p> <div class="language-console highlighter-rouge"><div class="highlight"><pre class="highlight"><code><span class="gp">$</span><span class="w"> </span>java <span class="nt">-jar</span> &lt;path_to_the_latest_pharmcat_jar&gt; <span class="nt">-phenotyper</span> <span class="nt">-pi</span> &lt;sample.match.json&gt;
</code></pre></div></div> <p>Sample script:</p> <div class="language-shell highlighter-rouge"><div class="highlight"><pre class="highlight"><code><span class="nb">cd </span>PharmCAT-tutorial/
<span class="k">for </span>SINGLE_SAMPLE <span class="k">in</span> <span class="s2">"</span><span class="si">$(</span><span class="nb">cat </span>data/test_get-rm_samples.txt<span class="si">)</span><span class="s2">"</span>
<span class="k">do
  </span>java <span class="nt">-jar</span> pharmcat.jar <span class="nt">-phenotyper</span> <span class="nt">-pi</span> results/pharmcat_ready/<span class="s2">"</span><span class="nv">$SINGLE_SAMPLE</span><span class="s2">"</span>.match.json
<span class="k">done</span>
</code></pre></div></div> <h4 id="reporter"> <a href="#reporter" class="anchor-heading" aria-labelledby="reporter"><svg viewBox="0 0 16 16" aria-hidden="true"><use xlink:href="#svg-link"></use></svg></a> Reporter </h4> <p>On the command line:</p> <div class="language-console highlighter-rouge"><div class="highlight"><pre class="highlight"><code><span class="gp">$</span><span class="w"> </span>java <span class="nt">-jar</span> &lt;path_to_the_latest_pharmcat_jar&gt; <span class="nt">-reporter</span> <span class="nt">-ri</span> &lt;sample.phenotype.json&gt; <span class="nt">-reporterJson</span>
</code></pre></div></div> <p>Sample script:</p> <div class="language-shell highlighter-rouge"><div class="highlight"><pre class="highlight"><code><span class="nb">cd </span>PharmCAT-tutorial/
<span class="k">for </span>SINGLE_SAMPLE <span class="k">in</span> <span class="s2">"</span><span class="si">$(</span><span class="nb">cat</span> &lt;sample_list.txt&gt;<span class="si">)</span><span class="s2">"</span>
<span class="k">do
  </span>java <span class="nt">-jar</span> pharmcat.jar <span class="nt">-reporter</span> <span class="nt">-ri</span> <span class="s2">"</span><span class="nv">$SINGLE_SAMPLE</span><span class="s2">"</span>.phenotype.json <span class="nt">-reporterJson</span> <span class="se">\</span>
    <span class="nt">-t</span> <span class="s1">'Report for '</span><span class="s2">"</span><span class="nv">$SINGLE_SAMPLE</span><span class="s2">"</span>
<span class="k">done</span>
</code></pre></div></div> <p>These commands yield <code class="language-plaintext highlighter-rouge">Named Allele Matcher</code>, <code class="language-plaintext highlighter-rouge">Phenotyper</code>, and <code class="language-plaintext highlighter-rouge">Reporter</code> results for each individual separately in the format of JSON files. We encourage the users to explore and perform data analysis using the rich content in these JSON files, which can be easily achieved using auxiliary JSON libraries and data analysis packages in R or python.</p> <h3 id="extracting-pharmcat-json-content-into-tsv"> <a href="#extracting-pharmcat-json-content-into-tsv" class="anchor-heading" aria-labelledby="extracting-pharmcat-json-content-into-tsv"><svg viewBox="0 0 16 16" aria-hidden="true"><use xlink:href="#svg-link"></use></svg></a> Extracting PharmCAT JSON content into TSV </h3> <p>We also provide accessory R scripts that organize and extract the content from the <a href="https://github.com/PharmGKB/PharmCAT-tutorial/blob/main/src/organize_pharmcat_named_allele_matcher_results.R">Named Allele Matcher</a> or <a href="https://github.com/PharmGKB/PharmCAT-tutorial/blob/main/src/organize_pharmcat_phenotyper_results.R">Phenotyper</a> JSON outputs into tab-separated values (TSV) files. Here is an example TSV that the users will obtain from the provided R scripts.</p> <div class="table-wrapper"><table> <thead> <tr> <th>samples</th> <th>gene</th> <th>diplotype</th> <th>haplotype_1</th> <th>haplotype_2</th> <th>haplotype_1_variants</th> <th>haplotype_2_variants</th> <th>missing_positions</th> </tr> </thead> <tbody> <tr> <td>NA18526</td> <td>ABCG2</td> <td>rs2231142 reference (G)/rs2231142 reference (G)</td> <td>rs2231142 reference (G)</td> <td>rs2231142 reference (G)</td> <td>88131171_G</td> <td>88131171_G</td> <td>NULL</td> </tr> <tr> <td>NA18526</td> <td>CYP3A5</td> <td>*1/*1</td> <td>*1</td> <td>*1</td> <td>99652770_T;99665212_C;99672916_T</td> <td>99652770_T;99665212_C;99672916_T</td> <td>99660516;99676198</td> </tr> <tr> <td>NA18526</td> <td>SLCO1B1</td> <td>*1/*15</td> <td>*1</td> <td>*15</td> <td>21176804_A;21176868_A;21176879_C;21176898_G;&lt;…truncated for visual clarity…&gt;</td> <td>21176804_G;21176868_A;21176879_C;&lt;…truncated for visual clarity…&gt;</td> <td>21172734;&lt;…truncated for visual clarity…&gt;</td> </tr> <tr> <td>NA18565</td> <td>ABCG2</td> <td>rs2231142 reference (G)/rs2231142 reference (G)</td> <td>rs2231142 reference (G)</td> <td>rs2231142 reference (G)</td> <td>88131171_G</td> <td>88131171_G</td> <td>NULL</td> </tr> <tr> <td>NA18565</td> <td>CYP3A5</td> <td>*1/*3</td> <td>*1</td> <td>*3</td> <td>99652770_T;99665212_C;99672916_T</td> <td>99652770_T;99665212_C;99672916_C</td> <td>99660516;99676198</td> </tr> <tr> <td>NA18565</td> <td>SLCO1B1</td> <td>*1/*1</td> <td>*1</td> <td>*1</td> <td>21176804_A;21176868_A;21176879_C;&lt;…truncated for visual clarity…&gt;</td> <td>21176804_A;21176868_A;21176879_C;&lt;…truncated for visual clarity…&gt;</td> <td>21172734;;&lt;…truncated for visual clarity…&gt;</td> </tr> <tr> <td>NA18861</td> <td>CYP3A5</td> <td>*1/*1</td> <td>*1</td> <td>*1</td> <td>99652770_T;99665212_C;99672916_T</td> <td>99652770_T;99665212_C;99672916_T</td> <td>99660516;99676198</td> </tr> <tr> <td>NA18861</td> <td>SLCO1B1</td> <td>*1/*43</td> <td>*1</td> <td>*43</td> <td>21176804_A;21176868_A;21176879_C;&lt;…truncated for visual clarity…&gt;</td> <td>21176804_G;21176868_A;21176879_A;&lt;…truncated for visual clarity…&gt;</td> <td>21172734;;&lt;…truncated for visual clarity…&gt;</td> </tr> </tbody> </table></div> <h4 id="extracting-the-pharmcat-named-allele-matcher-json-data-into-a-tsv-file"> <a href="#extracting-the-pharmcat-named-allele-matcher-json-data-into-a-tsv-file" class="anchor-heading" aria-labelledby="extracting-the-pharmcat-named-allele-matcher-json-data-into-a-tsv-file"><svg viewBox="0 0 16 16" aria-hidden="true"><use xlink:href="#svg-link"></use></svg></a> Extracting the PharmCAT Named Allele Matcher JSON data into a TSV file </h4> <div class="language-shell highlighter-rouge"><div class="highlight"><pre class="highlight"><code><span class="nb">cd </span>PharmCAT-tutorial/
<span class="nv">SCRIPT_PATH</span><span class="o">=</span>src/organize_pharmcat_named_allele_matcher_results.R
<span class="nv">MATCHER_DIR</span><span class="o">=</span>results/pharmcat_named_allele_matcher/
<span class="nv">MATCHER_PATTERN</span><span class="o">=</span>pharmcat_named_allele_matcher<span class="k">*</span>json
<span class="nv">PROJECT_DIR</span><span class="o">=</span><span class="s2">"</span><span class="nv">$PWD</span><span class="s2">"</span>
Rscript  “<span class="nv">$SCRIPT_PATH</span>” <span class="se">\</span>
<span class="nt">--input-dir</span> <span class="s2">"</span><span class="nv">$MATCHER_DIR</span><span class="s2">"</span> <span class="se">\</span>
<span class="nt">--input-file-pattern</span> <span class="s2">"</span><span class="nv">$MATCHER_PATTERN</span><span class="s2">"</span> <span class="se">\</span>
<span class="nt">--output-dir</span> <span class="s2">"</span><span class="nv">$PROJECT_DIR</span><span class="s2">"</span>
</code></pre></div></div> <h4 id="extracting-the-pharmcat-phenotyper-json-data-into-a-tsv-file"> <a href="#extracting-the-pharmcat-phenotyper-json-data-into-a-tsv-file" class="anchor-heading" aria-labelledby="extracting-the-pharmcat-phenotyper-json-data-into-a-tsv-file"><svg viewBox="0 0 16 16" aria-hidden="true"><use xlink:href="#svg-link"></use></svg></a> Extracting the PharmCAT Phenotyper JSON data into a TSV file </h4> <div class="language-shell highlighter-rouge"><div class="highlight"><pre class="highlight"><code><span class="nb">cd </span>PharmCAT-tutorial/
<span class="nv">SCRIPT_PATH</span><span class="o">=</span>src/organize_pharmcat_phenotyper_results.R
<span class="nv">PHENOTYPER_DIR</span><span class="o">=</span>results/pharmcat_phenotyper/
<span class="nv">PHENOTYPER_PATTERN</span><span class="o">=</span>pharmcat_phenotyper<span class="k">*</span>json
<span class="nv">PROJECT_DIR</span><span class="o">=</span><span class="s2">"</span><span class="nv">$PWD</span><span class="s2">"</span>
Rscript  <span class="s2">"</span><span class="nv">$SCRIPT_PATH</span><span class="s2">"</span> <span class="se">\</span>
<span class="nt">--input-dir</span> <span class="s2">"</span><span class="nv">$PHENOTYPER_DIR</span><span class="s2">"</span> <span class="se">\</span>
<span class="nt">--input-file-pattern</span> <span class="s2">"</span><span class="nv">$PHENOTYPER_PATTERN</span><span class="s2">"</span> <span class="se">\</span>
<span class="nt">--output-dir</span> <span class="s2">"</span><span class="nv">$PROJECT_DIR</span><span class="s2">"</span>
</code></pre></div></div><hr /> <h2 id="example-with-the-penn-medicine-biobank"> <a href="#example-with-the-penn-medicine-biobank" class="anchor-heading" aria-labelledby="example-with-the-penn-medicine-biobank"><svg viewBox="0 0 16 16" aria-hidden="true"><use xlink:href="#svg-link"></use></svg></a> Example with the Penn Medicine Biobank </h2> <p>This section provides an alternative example which is carried out on the Penn Medicine Biobank data. This example provides scripts that convert PharmCAT JSON files into a Python DataFrame and CSV/TSV</p> <p>The Stanford and UPenn PharmCAT teams independently developed methods for post-processing multiple samples from PharmCAT. This method, developed by UPenn, takes the output JSON files from multiple PharmCAT runs and converts them to a DataFrame object for use in Python or export to CSV/TSV.</p> <h3 id="preprocessing-the-data-1"> <a href="#preprocessing-the-data-1" class="anchor-heading" aria-labelledby="preprocessing-the-data-1"><svg viewBox="0 0 16 16" aria-hidden="true"><use xlink:href="#svg-link"></use></svg></a> Preprocessing the data </h3> <p>Before proceeding, you should preprocess your multi-sample VCF or single sample VCFs using <a href="/using/VCF-Preprocessor">PharmCAT's VCF preprocessor</a>.</p> <h3 id="running-pharmcat-1"> <a href="#running-pharmcat-1" class="anchor-heading" aria-labelledby="running-pharmcat-1"><svg viewBox="0 0 16 16" aria-hidden="true"><use xlink:href="#svg-link"></use></svg></a> Running PharmCAT </h3> <p>Running PharmCAT on the single-sample VCFs is pretty straightforward. First you need to create a text file with all of all the output file names from the preprocessor. This can easily be generated with the command:</p> <div class="language-console highlighter-rouge"><div class="highlight"><pre class="highlight"><code><span class="gp">$</span><span class="w"> </span><span class="nb">ls</span> &lt;preprocess_out_dir&gt; | <span class="nb">grep</span> <span class="s2">".vcf$"</span> <span class="o">&gt;</span> pharmcat_inputs.txt<span class="sb">`</span>
</code></pre></div></div> <p>Verify that this contains all the samples by taking the line count (<code class="language-plaintext highlighter-rouge">wc -l pharmcat_inputs.txt</code>) and ensuring it equals the number of samples.</p> <p>Then you need to use a for-loop to run PharmCAT repeatedly on each input file. The <code class="language-plaintext highlighter-rouge">-reporterJson</code> flag ensures that PharmCAT outputs JSON files for each sample containing the calls.</p> <div class="language-shell highlighter-rouge"><div class="highlight"><pre class="highlight"><code><span class="nv">BASE_DIR</span><span class="o">=</span><span class="s2">"~/project/preprocess_out"</span>  <span class="c"># the output directory from the preprocessor</span>
<span class="nv">OUT_DIR</span><span class="o">=</span><span class="s2">"pharmcat_out/"</span>  <span class="c"># where to put the PharmCAT output</span>
<span class="k">for </span>i <span class="k">in</span> <span class="si">$(</span><span class="nb">cat </span>pharmcat_inputs.txt<span class="si">)</span> <span class="c"># a file with the filenames of your preprocessed VCFs</span>
<span class="k">do
    </span><span class="nv">SAMPLE</span><span class="o">=</span><span class="sb">`</span><span class="nb">echo</span>  <span class="nv">$i</span> | <span class="nb">cut</span> <span class="nt">-d</span> <span class="s2">"."</span> <span class="nt">-f</span> 2<span class="sb">`</span> <span class="c"># the sample name is in the preprocessor output filename</span>
    <span class="c"># always use the latest version of PharmCAT</span>
    java <span class="nt">-jar</span> pharmcat.jar <span class="nt">-vcf</span> <span class="k">${</span><span class="nv">BASE_DIR</span><span class="k">}</span>/<span class="k">${</span><span class="nv">i</span><span class="k">}</span> <span class="nt">-bf</span> <span class="k">${</span><span class="nv">SAMPLE</span><span class="k">}</span> <span class="nt">-reporterJson</span> <span class="nt">-o</span> <span class="k">${</span><span class="nv">OUT_DIR</span><span class="k">}</span>
<span class="k">done</span>
</code></pre></div></div> <h3 id="reading-the-pharmcat-calls-into-python"> <a href="#reading-the-pharmcat-calls-into-python" class="anchor-heading" aria-labelledby="reading-the-pharmcat-calls-into-python"><svg viewBox="0 0 16 16" aria-hidden="true"><use xlink:href="#svg-link"></use></svg></a> Reading the PharmCAT calls into Python </h3> <p>Here is a exemplary python output for the PharmCAT Named Allele Matcher:</p> <div class="table-wrapper"><table> <thead> <tr> <th>SAMPLE_ID</th> <th>CACNA1S</th> <th>CFTR</th> <th>CYP2B6</th> <th>CYP2C19</th> <th>rs12777823</th> <th>CYP2C9</th> <th>&lt;…truncated for visual clarity…&gt;</th> </tr> </thead> <tbody> <tr> <td>SAMPLE_1</td> <td>Reference/Reference</td> <td>No CPIC variants found</td> <td>*1/*1</td> <td>*1/*1</td> <td>G/G</td> <td>*1/*1</td> <td>&lt;…truncated for visual clarity…&gt;</td> </tr> <tr> <td>SAMPLE_2</td> <td>Reference/Reference</td> <td>No CPIC variants found</td> <td>*1/*1</td> <td>*1/*1</td> <td>G/G</td> <td>*1/*1</td> <td>&lt;…truncated for visual clarity…&gt;</td> </tr> <tr> <td>SAMPLE_3</td> <td>Reference/Reference</td> <td>No CPIC variants found</td> <td>*1/*1</td> <td>*1/*1</td> <td>G/G</td> <td>*1/*1</td> <td>&lt;…truncated for visual clarity…&gt;</td> </tr> </tbody> </table></div> <p>Below is Python3 code for converting the JSON files from PharmCAT Named Allele Matcher and Phenotyper into Pandas DataFrames, which can be exported to CSV or processed further in python. Because of the long runtime of this process, we recommend saving the calls to CSV and loading the data from CSV when needed, instead of regenerating the DataFrames many times.</p> <div class="language-python highlighter-rouge"><div class="highlight"><pre class="highlight"><code><span class="c1"># Need to give a list of sample IDs
</span><span class="n">sample_list</span> <span class="o">=</span> <span class="p">[</span><span class="n">x</span><span class="p">.</span><span class="n">strip</span><span class="p">()</span> <span class="k">for</span> <span class="n">x</span> <span class="ow">in</span> <span class="nb">open</span><span class="p">(</span><span class="s">"sample_ids.txt"</span><span class="p">,</span> <span class="s">"r"</span><span class="p">).</span><span class="n">readlines</span><span class="p">()]</span>
<span class="c1"># PharmCAT output directory
</span><span class="n">base_dir</span> <span class="o">=</span> <span class="s">"~/project/pharmcat_out/"</span>

<span class="c1"># The following list specifies which genes should be pulled from the PharmCAT output. Be sure to include any new genes that get added to PharmCAT if they are relevant to you.
</span><span class="n">genelist</span> <span class="o">=</span> <span class="p">[</span><span class="s">"CACNA1S"</span><span class="p">,</span><span class="s">"CFTR"</span><span class="p">,</span><span class="s">"CYP2B6"</span><span class="p">,</span><span class="s">"CYP2C19"</span><span class="p">,</span><span class="s">"rs12777823"</span><span class="p">,</span><span class="s">"CYP2C9"</span><span class="p">,</span><span class="s">"CYP3A5"</span><span class="p">,</span><span class="s">"CYP4F2"</span><span class="p">,</span><span class="s">"DPYD"</span><span class="p">,</span><span class="s">"IFNL3/4"</span><span class="p">,</span><span class="s">"NUDT15"</span><span class="p">,</span><span class="s">"RYR1"</span><span class="p">,</span><span class="s">"TPMT"</span><span class="p">,</span><span class="s">"UGT1A1"</span><span class="p">,</span><span class="s">"VKORC1"</span><span class="p">,</span><span class="s">"SLCO1B1"</span><span class="p">]</span>
<span class="n">columns</span> <span class="o">=</span> <span class="p">[</span><span class="s">"PID"</span><span class="p">]</span> <span class="o">+</span> <span class="n">genelist</span>


<span class="k">for</span> <span class="n">pid</span> <span class="ow">in</span> <span class="n">sample_list</span><span class="p">:</span>
    <span class="n">pid</span> <span class="o">=</span> <span class="nb">str</span><span class="p">(</span><span class="n">pid</span><span class="p">)</span>
    <span class="c1"># Load in the JSON
</span>    <span class="n">sample_file</span> <span class="o">=</span> <span class="s">"{0}/{1}.report.json"</span><span class="p">.</span><span class="nb">format</span><span class="p">(</span><span class="n">base_dir</span><span class="p">,</span> <span class="n">pid</span><span class="p">.</span><span class="n">strip</span><span class="p">())</span>
    <span class="n">sample_json</span> <span class="o">=</span> <span class="n">json</span><span class="p">.</span><span class="n">load</span><span class="p">(</span><span class="nb">open</span><span class="p">(</span><span class="n">sample_file</span><span class="p">,</span> <span class="s">"r"</span><span class="p">))</span>

    <span class="c1"># Build a dictionary that stores the genotype and phenotype for each gene from the JSON
</span>    <span class="n">gene2genotype</span> <span class="o">=</span> <span class="nb">dict</span><span class="p">([(</span><span class="n">key</span><span class="p">,</span> <span class="s">"NA"</span><span class="p">)</span> <span class="k">for</span> <span class="n">key</span> <span class="ow">in</span> <span class="n">genelist</span><span class="p">])</span>
    <span class="n">gene2phenotype</span> <span class="o">=</span> <span class="nb">dict</span><span class="p">([(</span><span class="n">key</span><span class="p">,</span> <span class="s">"NA"</span><span class="p">)</span> <span class="k">for</span> <span class="n">key</span> <span class="ow">in</span> <span class="n">genelist</span><span class="p">])</span>
    <span class="k">for</span> <span class="n">genotype</span> <span class="ow">in</span> <span class="n">sample_json</span><span class="p">[</span><span class="s">"genotypes"</span><span class="p">]:</span>
        <span class="n">gene</span> <span class="o">=</span> <span class="n">genotype</span><span class="p">[</span><span class="s">"gene"</span><span class="p">]</span>
        <span class="c1"># since there can be multiple genotypes called due to ambiguity, join them with ';'
</span>        <span class="n">gene2genotype</span><span class="p">[</span><span class="n">gene</span><span class="p">]</span> <span class="o">=</span> <span class="s">";"</span><span class="p">.</span><span class="n">join</span><span class="p">(</span><span class="nb">set</span><span class="p">(</span><span class="n">genotype</span><span class="p">[</span><span class="s">"calls"</span><span class="p">]))</span>
        <span class="n">gene2phenotype</span><span class="p">[</span><span class="n">gene</span><span class="p">]</span> <span class="o">=</span> <span class="s">";"</span><span class="p">.</span><span class="n">join</span><span class="p">(</span><span class="nb">set</span><span class="p">(</span><span class="n">genotype</span><span class="p">[</span><span class="s">"phenotype"</span><span class="p">]))</span>

    <span class="c1"># Optional section: code to extract rs12777823 for Warfarin (remove rs12777823 from genelist if deleting this section)
</span>    <span class="k">for</span> <span class="n">call</span> <span class="ow">in</span> <span class="n">sample_json</span><span class="p">[</span><span class="s">"geneCalls"</span><span class="p">]:</span>
        <span class="k">if</span> <span class="n">call</span><span class="p">[</span><span class="s">"gene"</span><span class="p">]</span> <span class="o">==</span> <span class="s">"CYP2C9"</span><span class="p">:</span>
            <span class="k">for</span> <span class="n">variant</span> <span class="ow">in</span> <span class="n">call</span><span class="p">[</span><span class="s">"variantsOfInterest"</span><span class="p">]:</span>
                <span class="k">if</span> <span class="n">variant</span><span class="p">[</span><span class="s">"dbSnpId"</span><span class="p">]</span> <span class="o">==</span> <span class="s">"rs12777823"</span><span class="p">:</span>
                    <span class="n">gene2genotype</span><span class="p">[</span><span class="s">"rs12777823"</span><span class="p">]</span> <span class="o">=</span> <span class="n">variant</span><span class="p">[</span><span class="s">'call'</span><span class="p">]</span>
                    <span class="k">if</span> <span class="n">gene2genotype</span><span class="p">[</span><span class="s">"rs12777823"</span><span class="p">]</span> <span class="ow">is</span> <span class="bp">None</span><span class="p">:</span>
                        <span class="n">gene2genotype</span><span class="p">[</span><span class="s">"rs12777823"</span><span class="p">]</span> <span class="o">=</span> <span class="s">"NA"</span>
                    <span class="n">gene2genotype</span><span class="p">[</span><span class="s">"rs12777823"</span><span class="p">]</span> <span class="o">=</span> <span class="n">gene2genotype</span><span class="p">[</span><span class="s">"rs12777823"</span><span class="p">].</span><span class="n">replace</span><span class="p">(</span><span class="s">"|"</span><span class="p">,</span> <span class="s">"/"</span><span class="p">)</span>

    <span class="c1"># Convert the dictionaries into DataFrames
</span>	<span class="n">genotypes</span> <span class="o">=</span> <span class="n">pd</span><span class="p">.</span><span class="n">DataFrame</span><span class="p">(</span><span class="n">columns</span><span class="o">=</span><span class="n">columns</span><span class="p">)</span>
	<span class="n">phenotypes</span> <span class="o">=</span> <span class="n">pd</span><span class="p">.</span><span class="n">DataFrame</span><span class="p">(</span><span class="n">columns</span><span class="o">=</span><span class="n">columns</span><span class="p">)</span>
    <span class="c1">## Convert the dictionary into lists corresponding to DataFrame rows (one per sample)
</span>    <span class="n">geno_row</span> <span class="o">=</span> <span class="p">[</span><span class="n">pid</span><span class="p">.</span><span class="n">strip</span><span class="p">()]</span> <span class="o">+</span> <span class="p">[</span><span class="n">gene2genotype</span><span class="p">[</span><span class="n">gene</span><span class="p">]</span> <span class="k">for</span> <span class="n">gene</span> <span class="ow">in</span> <span class="n">gene2genotype</span><span class="p">]</span>
    <span class="n">pheno_row</span> <span class="o">=</span> <span class="p">[</span><span class="n">pid</span><span class="p">.</span><span class="n">strip</span><span class="p">()]</span> <span class="o">+</span> <span class="p">[</span><span class="n">gene2phenotype</span><span class="p">[</span><span class="n">gene</span><span class="p">]</span> <span class="k">for</span> <span class="n">gene</span> <span class="ow">in</span> <span class="n">gene2phenotype</span><span class="p">]</span>

    <span class="c1">## Add each row to the DataFrames
</span>    <span class="n">genotypes</span> <span class="o">=</span> <span class="n">genotypes</span><span class="p">.</span><span class="n">append</span><span class="p">(</span><span class="n">pd</span><span class="p">.</span><span class="n">Series</span><span class="p">(</span><span class="n">geno_row</span><span class="p">,</span> <span class="n">index</span><span class="o">=</span><span class="n">genotypes</span><span class="p">.</span><span class="n">columns</span><span class="p">),</span> <span class="n">ignore_index</span><span class="o">=</span><span class="bp">True</span><span class="p">)</span>
    <span class="n">phenotypes</span> <span class="o">=</span> <span class="n">phenotypes</span><span class="p">.</span><span class="n">append</span><span class="p">(</span><span class="n">pd</span><span class="p">.</span><span class="n">Series</span><span class="p">(</span><span class="n">pheno_row</span><span class="p">,</span> <span class="n">index</span><span class="o">=</span><span class="n">phenotypes</span><span class="p">.</span><span class="n">columns</span><span class="p">),</span> <span class="n">ignore_index</span><span class="o">=</span><span class="bp">True</span><span class="p">)</span>

    <span class="c1"># Optional section: re-encode phenotypes from Warfarin-related genes
</span>    <span class="n">phenotypes</span><span class="p">[</span><span class="s">"CYP4F2"</span><span class="p">]</span> <span class="o">=</span> <span class="n">genotypes</span><span class="p">[</span><span class="s">"CYP4F2"</span><span class="p">].</span><span class="nb">apply</span><span class="p">(</span><span class="k">lambda</span> <span class="n">x</span><span class="p">:</span> <span class="s">"*3"</span> <span class="ow">in</span> <span class="n">x</span><span class="p">).</span><span class="n">replace</span><span class="p">({</span><span class="bp">False</span><span class="p">:</span><span class="s">"not actionable"</span><span class="p">,</span> <span class="bp">True</span><span class="p">:</span><span class="s">"actionable"</span><span class="p">})</span>
    <span class="n">phenotypes</span><span class="p">[</span><span class="s">"VKORC1"</span><span class="p">]</span> <span class="o">=</span> <span class="n">genotypes</span><span class="p">[</span><span class="s">"VKORC1"</span><span class="p">].</span><span class="nb">apply</span><span class="p">(</span><span class="k">lambda</span> <span class="n">x</span><span class="p">:</span> <span class="s">"variant"</span> <span class="ow">in</span> <span class="n">x</span><span class="p">).</span><span class="n">replace</span><span class="p">({</span><span class="bp">False</span><span class="p">:</span><span class="s">"not actionable"</span><span class="p">,</span> <span class="bp">True</span><span class="p">:</span><span class="s">"actionable"</span><span class="p">})</span>
    <span class="n">phenotypes</span><span class="p">[</span><span class="s">"IFNL3/4"</span><span class="p">]</span> <span class="o">=</span> <span class="n">genotypes</span><span class="p">[</span><span class="s">"IFNL3/4"</span><span class="p">].</span><span class="nb">apply</span><span class="p">(</span><span class="k">lambda</span> <span class="n">x</span><span class="p">:</span> <span class="s">"variant"</span> <span class="ow">in</span> <span class="n">x</span><span class="p">).</span><span class="n">replace</span><span class="p">({</span><span class="bp">False</span><span class="p">:</span><span class="s">"not actionable"</span><span class="p">,</span> <span class="bp">True</span><span class="p">:</span><span class="s">"actionable"</span><span class="p">})</span>
    <span class="n">phenotypes</span><span class="p">[</span><span class="s">"rs12777823"</span><span class="p">]</span> <span class="o">=</span> <span class="n">genotypes</span><span class="p">[</span><span class="s">"rs12777823"</span><span class="p">].</span><span class="nb">apply</span><span class="p">(</span><span class="k">lambda</span> <span class="n">x</span><span class="p">:</span> <span class="s">"A"</span> <span class="ow">in</span> <span class="n">x</span><span class="p">).</span><span class="n">replace</span><span class="p">({</span><span class="bp">False</span><span class="p">:</span><span class="s">"not actionable"</span><span class="p">,</span> <span class="bp">True</span><span class="p">:</span><span class="s">"actionable"</span><span class="p">})</span>

<span class="n">genotypes</span> <span class="o">=</span> <span class="n">genotypes</span><span class="p">.</span><span class="n">set_index</span><span class="p">(</span><span class="s">"PID"</span><span class="p">)</span>
<span class="n">phenotypes</span> <span class="o">=</span> <span class="n">phenotypes</span><span class="p">.</span><span class="n">set_index</span><span class="p">(</span><span class="s">"PID"</span><span class="p">)</span>

<span class="c1"># Output the DataFrames as CSV
</span><span class="n">genotypes</span><span class="p">.</span><span class="n">to_csv</span><span class="p">(</span><span class="s">"genotypes.csv"</span><span class="p">)</span>
<span class="n">phenotypes</span><span class="p">.</span><span class="n">to_csv</span><span class="p">(</span><span class="s">"phenotypes.csv"</span><span class="p">)</span>
</code></pre></div></div> <p>The <code class="language-plaintext highlighter-rouge">genotypes</code> and <code class="language-plaintext highlighter-rouge">phenotypes</code> DataFrames should have the sample ID as the index, with one sample per row and one gene per column.</p><hr /> <h2 id="reference"> <a href="#reference" class="anchor-heading" aria-labelledby="reference"><svg viewBox="0 0 16 16" aria-hidden="true"><use xlink:href="#svg-link"></use></svg></a> Reference </h2> <ol> <li>Pratt, V. M. et al. Characterization of 137 Genomic DNA Reference Materials for 28 Pharmacogenetic Genes: A GeT-RM Collaborative Project. J Mol Diagn 18, 109–123 (2016).</li> <li>Byrska-Bishop, M. et al. High coverage whole genome sequencing of the expanded 1000 Genomes Project cohort including 602 trios. 2021.02.06.430068 (2021) doi:10.1101/2021.02.06.430068.</li> </ol> <hr> <footer> <div> PharmCAT is managed at Stanford University & University of Pennsylvania (NHGRI U24HG010862). </div> </footer> </div> </div> <div class="search-overlay"></div> </div> </body> </html>
