<!DOCTYPE html> <html lang="en-US"> <head> <meta charset="UTF-8"> <meta http-equiv="X-UA-Compatible" content="IE=Edge"> <title>Running PharmCAT - PharmCAT</title> <link rel="shortcut icon" href="/favicon.ico" type="image/x-icon"> <link rel="stylesheet" href="/assets/css/just-the-docs-default.css"> <script async src="https://www.googletagmanager.com/gtag/js?id=G-7ZP1HHBWTE"></script> <script> window.dataLayer = window.dataLayer || []; function gtag(){dataLayer.push(arguments);} gtag('js', new Date()); gtag('config', 'G-7ZP1HHBWTE', { 'anonymize_ip': true }); </script> <script type="text/javascript" src="/assets/js/vendor/lunr.min.js"></script> <script type="text/javascript" src="/assets/js/just-the-docs.js"></script> <meta name="viewport" content="width=device-width, initial-scale=1"> <!-- Begin Jekyll SEO tag v2.8.0 --> <title>Running PharmCAT | PharmCAT</title> <meta name="generator" content="Jekyll v4.2.2" /> <meta property="og:title" content="Running PharmCAT" /> <meta property="og:locale" content="en_US" /> <meta name="description" content="The Pharmacogenomics Clinical Annotation Tool (PharmCAT) is an open-source tool that takes an individual’s genetic data and determines their respective drug-related genotypes and phenotypes. PharmCAT produces a personalized report with genotype-specific expert-reviewed drug prescribing recommendations." /> <meta property="og:description" content="The Pharmacogenomics Clinical Annotation Tool (PharmCAT) is an open-source tool that takes an individual’s genetic data and determines their respective drug-related genotypes and phenotypes. PharmCAT produces a personalized report with genotype-specific expert-reviewed drug prescribing recommendations." /> <meta property="og:site_name" content="PharmCAT" /> <meta property="og:type" content="website" /> <meta name="twitter:card" content="summary" /> <meta property="twitter:title" content="Running PharmCAT" /> <script type="application/ld+json"> {"@context":"https://schema.org","@type":"WebPage","description":"The Pharmacogenomics Clinical Annotation Tool (PharmCAT) is an open-source tool that takes an individual’s genetic data and determines their respective drug-related genotypes and phenotypes. PharmCAT produces a personalized report with genotype-specific expert-reviewed drug prescribing recommendations.","headline":"Running PharmCAT","publisher":{"@type":"Organization","logo":{"@type":"ImageObject","url":"/images/pharmcat_header.png"}},"url":"/using/Running-PharmCAT.html"}</script> <!-- End Jekyll SEO tag --> <script async defer data-website-id="714c506c-5f1e-4ea8-af79-fa80c02911c6" src="https://metrics.cpicpgx.org/j.js"></script> <style> h4 { font-size: 14px !important; font-weight: bold; text-transform: none; } h4 code { font-size: 0.9em; font-weight: bold; } h5 code { font-size: .9em; } .main-content ul, .main-content ol { padding-left: 2em; } .main-content dl { grid-template-columns: 16em 1fr; } .main-content dt::after { content: unset; } .altArg { font-size: .8em; font-weight: normal; } footer { position: relative; top: -1.5em; font-size: smaller; } </style> <script src="https://cdnjs.cloudflare.com/ajax/libs/mermaid/9.1.1/mermaid.min.js" integrity="sha512-7AtDHIWTEL1Km2V9clac6gs1x3F67rS6HGZMBYaUogsohvODnWrq5GkW9WRh1XjTMs4Um9hjGUNSjI+3XluoZQ==" crossorigin="anonymous" referrerpolicy="no-referrer"></script> <script> function docReady(fn) { if (document.readyState === "complete" || document.readyState === "interactive") { setTimeout(fn, 1); } else { document.addEventListener("DOMContentLoaded", fn); } } docReady(function() { mermaid.initialize({ theme: 'base', themeVariables: { background: '#fff', primaryColor: '#fcfcfc', primaryBorderColor: '#999', fontFamily: 'system-ui, -apple-system, BlinkMacSystemFont, "Segoe UI", Roboto, "Helvetica Neue", Arial, sans-serif', fontSize: '14px', } }); }); </script> <style> .label { text-transform: unset; font-size: 14px !important; } .module > rect { stroke: #f00 !important; fill: #fffcfc !important; } .optional > rect { stroke: #ccc !important; } .optional > polygon { stroke: #ccc !important; } </style> </head> <body> <svg xmlns="http://www.w3.org/2000/svg" style="display: none;"> <symbol id="svg-link" viewBox="0 0 24 24"> <title>Link</title> <svg xmlns="http://www.w3.org/2000/svg" width="24" height="24" viewBox="0 0 24 24" fill="none" stroke="currentColor" stroke-width="2" stroke-linecap="round" stroke-linejoin="round" class="feather feather-link"> <path d="M10 13a5 5 0 0 0 7.54.54l3-3a5 5 0 0 0-7.07-7.07l-1.72 1.71"></path><path d="M14 11a5 5 0 0 0-7.54-.54l-3 3a5 5 0 0 0 7.07 7.07l1.71-1.71"></path> </svg> </symbol> <symbol id="svg-search" viewBox="0 0 24 24"> <title>Search</title> <svg xmlns="http://www.w3.org/2000/svg" width="24" height="24" viewBox="0 0 24 24" fill="none" stroke="currentColor" stroke-width="2" stroke-linecap="round" stroke-linejoin="round" class="feather feather-search"> <circle cx="11" cy="11" r="8"></circle><line x1="21" y1="21" x2="16.65" y2="16.65"></line> </svg> </symbol> <symbol id="svg-menu" viewBox="0 0 24 24"> <title>Menu</title> <svg xmlns="http://www.w3.org/2000/svg" width="24" height="24" viewBox="0 0 24 24" fill="none" stroke="currentColor" stroke-width="2" stroke-linecap="round" stroke-linejoin="round" class="feather feather-menu"> <line x1="3" y1="12" x2="21" y2="12"></line><line x1="3" y1="6" x2="21" y2="6"></line><line x1="3" y1="18" x2="21" y2="18"></line> </svg> </symbol> <symbol id="svg-arrow-right" viewBox="0 0 24 24"> <title>Expand</title> <svg xmlns="http://www.w3.org/2000/svg" width="24" height="24" viewBox="0 0 24 24" fill="none" stroke="currentColor" stroke-width="2" stroke-linecap="round" stroke-linejoin="round" class="feather feather-chevron-right"> <polyline points="9 18 15 12 9 6"></polyline> </svg> </symbol> <symbol id="svg-doc" viewBox="0 0 24 24"> <title>Document</title> <svg xmlns="http://www.w3.org/2000/svg" width="24" height="24" viewBox="0 0 24 24" fill="none" stroke="currentColor" stroke-width="2" stroke-linecap="round" stroke-linejoin="round" class="feather feather-file"> <path d="M13 2H6a2 2 0 0 0-2 2v16a2 2 0 0 0 2 2h12a2 2 0 0 0 2-2V9z"></path><polyline points="13 2 13 9 20 9"></polyline> </svg> </symbol> </svg> <div class="side-bar"> <div class="site-header"> <a href="/" class="site-title lh-tight"> <div class="site-logo"></div> </a> <a href="#" id="menu-button" class="site-button"> <svg viewBox="0 0 24 24" class="icon"><use xlink:href="#svg-menu"></use></svg> </a> </div> <nav role="navigation" aria-label="Main" id="site-nav" class="site-nav"> <ul class="nav-list"><li class="nav-list-item"><a href="/" class="nav-list-link">Home</a></li><li class="nav-list-item"><a href="#" class="nav-list-expander"><svg viewBox="0 0 24 24"><use xlink:href="#svg-arrow-right"></use></svg></a><a href="/methods/" class="nav-list-link">Methods</a><ul class="nav-list "><li class="nav-list-item "><a href="/methods/Modules.html" class="nav-list-link">PharmCAT Modules</a></li><li class="nav-list-item "><a href="/methods/determining-alleles" class="nav-list-link">Determining Alleles</a></li><li class="nav-list-item "><a href="/methods/NamedAlleleMatcher-101.html" class="nav-list-link">Named Allele Matcher 101</a></li><li class="nav-list-item "><a href="/methods/NamedAlleleMatcher-201.html" class="nav-list-link">Named Allele Matcher 201</a></li><li class="nav-list-item "><a href="/methods/gene-definition-exceptions" class="nav-list-link">Gene Definition Exceptions</a></li><li class="nav-list-item "><a href="/methods/versioning/" class="nav-list-link">Versioning</a></li></ul></li><li class="nav-list-item active"><a href="#" class="nav-list-expander"><svg viewBox="0 0 24 24"><use xlink:href="#svg-arrow-right"></use></svg></a><a href="/using/" class="nav-list-link">Using PharmCAT</a><ul class="nav-list "><li class="nav-list-item active"><a href="/using/Running-PharmCAT.html" class="nav-list-link active">Running PharmCAT</a></li><li class="nav-list-item "><a href="/using/VCF-Preprocessor.html" class="nav-list-link">VCF Preprocessor</a></li><li class="nav-list-item "><a href="/using/PharmCAT-in-Docker.html" class="nav-list-link">PharmCAT in Docker</a></li><li class="nav-list-item "><a href="/using/Multi-Sample-Analysis.html" class="nav-list-link">Multi-Sample Analysis</a></li><li class="nav-list-item "><a href="/using/Calling-CYP2D6.html" class="nav-list-link">Calling CYP2D6</a></li></ul></li><li class="nav-list-item"><a href="#" class="nav-list-expander"><svg viewBox="0 0 24 24"><use xlink:href="#svg-arrow-right"></use></svg></a><a href="/specifications/" class="nav-list-link">Specifications</a><ul class="nav-list "><li class="nav-list-item "><a href="/specifications/VCF-Requirements.html" class="nav-list-link">VCF Requirements</a></li><li class="nav-list-item "><a href="/specifications/preparing-vcf" class="nav-list-link">Preparing VCF</a></li><li class="nav-list-item "><a href="/specifications/Outside-Call-Format.html" class="nav-list-link">Outside Call Format</a></li></ul></li><li class="nav-list-item"><a href="/genes-drugs/" class="nav-list-link">Genes & Drugs</a></li><li class="nav-list-item"><a href="/examples/" class="nav-list-link">Examples</a></li><li class="nav-list-item"><a href="/faqs/" class="nav-list-link">FAQs</a></li><li class="nav-list-item"><a href="/disclaimers/" class="nav-list-link">Disclaimers and Other Information</a></li><li class="nav-list-item"><a href="/about-us/" class="nav-list-link">About Us</a></li></ul> </nav> <footer class="site-footer"> This site uses <a href="https://github.com/pmarsceill/just-the-docs">Just the Docs</a>, a documentation theme for Jekyll. </footer> </div> <div class="main" id="top"> <div id="main-header" class="main-header"> <div class="search"> <div class="search-input-wrap"> <input type="text" id="search-input" class="search-input" tabindex="0" placeholder="Search PharmCAT" aria-label="Search PharmCAT" autocomplete="off"> <label for="search-input" class="search-label"><svg viewBox="0 0 24 24" class="search-icon"><use xlink:href="#svg-search"></use></svg></label> </div> <div id="search-results" class="search-results"></div> </div> </div> <div id="main-content-wrap" class="main-content-wrap"> <nav aria-label="Breadcrumb" class="breadcrumb-nav"> <ol class="breadcrumb-nav-list"> <li class="breadcrumb-nav-list-item"><a href="/using/">Using PharmCAT</a></li> <li class="breadcrumb-nav-list-item"><span>Running PharmCAT</span></li> </ol> </nav> <div id="main-content" class="main-content" role="main"> <h1 id="running-pharmcat"> <a href="#running-pharmcat" class="anchor-heading" aria-labelledby="running-pharmcat"><svg viewBox="0 0 16 16" aria-hidden="true"><use xlink:href="#svg-link"></use></svg></a> Running PharmCAT </h1> <p>This document teaches you how to use the core PharmCAT tool.</p> <h2 id="requirements"> <a href="#requirements" class="anchor-heading" aria-labelledby="requirements"><svg viewBox="0 0 16 16" aria-hidden="true"><use xlink:href="#svg-link"></use></svg></a> Requirements </h2> <h3 id="install-the-software"> <a href="#install-the-software" class="anchor-heading" aria-labelledby="install-the-software"><svg viewBox="0 0 16 16" aria-hidden="true"><use xlink:href="#svg-link"></use></svg></a> Install the Software </h3> <p>You can skip this if are <a href="PharmCAT-in-Docker">running PharmCAT in Docker</a>.</p> <ol> <li>You will need <a href="https://adoptium.net/index.html?variant=openjdk17&amp;jvmVariant=hotspot">Java 17 or newer</a>.</li> <li>Download the PharmCAT Jar file from our <a href="https://github.com/PharmGKB/PharmCAT/releases/">releases page</a>.</li> </ol> <h3 id="prepare-your-vcf-files"> <a href="#prepare-your-vcf-files" class="anchor-heading" aria-labelledby="prepare-your-vcf-files"><svg viewBox="0 0 16 16" aria-hidden="true"><use xlink:href="#svg-link"></use></svg></a> Prepare Your VCF Files </h3> <p>PharmCAT takes VCF files as input.</p> <blockquote class="warn"> <p>Please make sure you have read and undertand PharmCAT's <a href="/specifications/VCF-Requirements">VCF requirements</a>.</p> <p>If you are not preparing your VCF files yourself, we highly recommend you run your VCF file through PharmCAT's <a href="/using/VCF-Preprocessor">VCF preprocessor</a>.</p> </blockquote> <h2 id="running-pharmcat-1"> <a href="#running-pharmcat-1" class="anchor-heading" aria-labelledby="running-pharmcat-1"><svg viewBox="0 0 16 16" aria-hidden="true"><use xlink:href="#svg-link"></use></svg></a> Running PharmCAT </h2> <p>This is the basic command, which will take your VCF file and produce the PharmCAT report:</p> <div class="language-console highlighter-rouge"><div class="highlight"><pre class="highlight"><code><span class="gp">#</span><span class="w"> </span>java <span class="nt">-jar</span> &lt;path_to_pharmcat_jar_file&gt; <span class="nt">-vcf</span> &lt;vcf_file&gt;
</code></pre></div></div> <p>Where:</p> <dl> <dt>-jar <code class="language-plaintext highlighter-rouge">&lt;path_to_jar_file&gt;</code></dt> <dd>The compiled PharmCAT Jar file</dd> <dt>-vcf <code class="language-plaintext highlighter-rouge">&lt;vcf_file&gt;</code></dt> <dd>Input VCF file (must comply with PharmCAT's <a href="/specifications/VCF-Requirements">VCF requirements</a>)</dd> </dl> <p>By default, the output will be saved to the same directory as the input VCF file and will use the same base file name. For example:</p> <div class="language-console highlighter-rouge"><div class="highlight"><pre class="highlight"><code><span class="gp">#</span><span class="w"> </span>java <span class="nt">-jar</span> pharmcat.jar <span class="nt">-vcf</span> /tmp/sample1.vcf
<span class="go">Saving named allele matcher JSON results to /tmp/sample1.match.json
Saving phenotyper JSON results to /tmp/sample1.phenotype.json
Saving reporter HTML results to /tmp/sample1.report.html
</span></code></pre></div></div> <p>Notice that all intermediary files will be kept.</p> <p>To control this behavior, provide:</p> <dl> <dt>-o <code class="language-plaintext highlighter-rouge">&lt;dir&gt;</code> <span class="altArg"><br />or --output-dir <code class="language-plaintext highlighter-rouge">&lt;dir&gt;</code></span></dt> <dd>Directory to save files to</dd> <dt>-bf <code class="language-plaintext highlighter-rouge">&lt;name&gt;</code> <span class="altArg"><br />or --base-filename <code class="language-plaintext highlighter-rouge">&lt;name&gt;</code></span></dt> <dd>The base name (without file extensions) used for output files</dd> <dt>-del <span class="altArg"><br />or --delete-intermediary-files</span></dt> <dd>Delete intermediary output files</dd> </dl> <p>Example:</p> <div class="language-console highlighter-rouge"><div class="highlight"><pre class="highlight"><code><span class="gp">#</span><span class="w"> </span>java <span class="nt">-jar</span> pharmcat.jar <span class="nt">-vcf</span> /tmp/input.vcf <span class="nt">-o</span> /tmp/results <span class="nt">-bf</span> sample
<span class="go">Saving named allele matcher JSON results to /tmp/results/sample.match.json
Saving phenotyper JSON results to /tmp/results/sample.phenotype.json
Saving reporter HTML results to /tmp/results/sample.report.html

</span><span class="gp">#</span><span class="w"> </span>java <span class="nt">-jar</span> pharmcat.jar <span class="nt">-vcf</span> /tmp/input.vcf <span class="nt">--output-dir</span> /tmp/results <span class="nt">-del</span>
<span class="go">Saving reporter HTML results to /tmp/results/input.report.html
</span></code></pre></div></div> <h3 id="outside-calls"> <a href="#outside-calls" class="anchor-heading" aria-labelledby="outside-calls"><svg viewBox="0 0 16 16" aria-hidden="true"><use xlink:href="#svg-link"></use></svg></a> Outside Calls </h3> <p>If you need to provide diplotypes directly to PharmCAT, you can do so using an <a href="/specifications/Outside-Call-Format">"outside calls" file</a>. You might want to do this for genes that PharmCAT does not call directly, or to override PharmCAT's call.</p> <p>To do so, provide:</p> <dl> <dt>-po <code class="language-plaintext highlighter-rouge">&lt;tsv_file&gt;</code> <span class="altArg"><br />or --phenotyper-outside-call-file <code class="language-plaintext highlighter-rouge">&lt;tsv_file&gt;</code></span></dt> <dd>Path to an outside call file (TSV)</dd> </dl> <p>Example:</p> <div class="language-console highlighter-rouge"><div class="highlight"><pre class="highlight"><code><span class="gp">#</span><span class="w"> </span>java <span class="nt">-jar</span> pharmcat.jar <span class="nt">-vcf</span> /tmp/sample.vcf <span class="nt">-po</span> /tmp/outside_calls.tsv
<span class="go">Saving named allele matcher JSON results to /tmp/results/sample.match.json
Saving phenotyper JSON results to /tmp/results/sample.phenotype.json
Saving reporter HTML results to /tmp/results/sample.report.html
</span></code></pre></div></div> <h2 id="advanced-usage"> <a href="#advanced-usage" class="anchor-heading" aria-labelledby="advanced-usage"><svg viewBox="0 0 16 16" aria-hidden="true"><use xlink:href="#svg-link"></use></svg></a> Advanced Usage </h2> <p>Remember that the PharmCAT tool is composed of 3 modules: the <code class="language-plaintext highlighter-rouge">Named Allele Matcher</code>, the <code class="language-plaintext highlighter-rouge">Phenotyper</code>, and the <code class="language-plaintext highlighter-rouge">Reporter</code>.</p> <div class="mermaid"> graph LR; MI{{vcf file}} --&gt; M[Named Allele Matcher]; M --&gt; P[Phenotyper]; PI{{"outside call file"}} -. optional .-&gt; P; P --&gt; R[Reporter]; R --&gt; RO([HTML report]); class M,P,R module; class PI optional; </div> <p>Each module has its own arguments to customize its behavior.</p> <h4 id="named-allele-matcher"> <a href="#named-allele-matcher" class="anchor-heading" aria-labelledby="named-allele-matcher"><svg viewBox="0 0 16 16" aria-hidden="true"><use xlink:href="#svg-link"></use></svg></a> Named Allele Matcher </h4> <dl> <dt>-matcher</dt> <dd>run Named Allele Matcher</dd> <dt>-ma <span class="altArg"><br />or --matcher-all-results</span></dt> <dd>return all possible diplotypes, not just top hits</dd> <dt>-md <code class="language-plaintext highlighter-rouge">&lt;dir&gt;</code> <span class="altArg"><br />or --matcher-definitions-dir <code class="language-plaintext highlighter-rouge">&lt;dir&gt;</code></span></dt> <dd>directory containing named allele definitions (JSON files)</dd> <dt>-matcherHtml <span class="altArg"><br />or --matcher-save-html</span></dt> <dd>save named allele matcher results as HTML</dd> </dl> <h4 id="phenotyper"> <a href="#phenotyper" class="anchor-heading" aria-labelledby="phenotyper"><svg viewBox="0 0 16 16" aria-hidden="true"><use xlink:href="#svg-link"></use></svg></a> Phenotyper </h4> <dl> <dt>-phenotyper</dt> <dd>run Phenotyper</dd> <dt>-pi <code class="language-plaintext highlighter-rouge">&lt;json_file&gt;</code> <span class="altArg"><br />or --phenotyper-input <code class="language-plaintext highlighter-rouge">&lt;json_file&gt;</code></span></dt> <dd>JSON results from named allele matcher</dd> <dt>-po <code class="language-plaintext highlighter-rouge">&lt;tsv_file&gt;</code> <span class="altArg"><br />or --phenotyper-outside-call-file <code class="language-plaintext highlighter-rouge">&lt;tsv_file&gt;</code></span></dt> <dd>path to an outside call file (TSV)</dd> </dl> <h4 id="reporter"> <a href="#reporter" class="anchor-heading" aria-labelledby="reporter"><svg viewBox="0 0 16 16" aria-hidden="true"><use xlink:href="#svg-link"></use></svg></a> Reporter </h4> <dl> <dt>-reporter</dt> <dd>run Reporter</dd> <dt>-ri <code class="language-plaintext highlighter-rouge">&lt;json_file&gt;</code> <span class="altArg"><br />or --reporter-input <code class="language-plaintext highlighter-rouge">&lt;json_file&gt;</code></span></dt> <dd>JSON results from phenotyper</dd> <dt>-rt <code class="language-plaintext highlighter-rouge">&lt;title&gt;</code> <span class="altArg"><br />or --reporter-title <code class="language-plaintext highlighter-rouge">&lt;title&gt;</code></span></dt> <dd>text to add to the report title</dd> <dt>-reporterJson</dt> <dd>save reporter results as JSON</dd> </dl> <h3 id="running-individual-modules"> <a href="#running-individual-modules" class="anchor-heading" aria-labelledby="running-individual-modules"><svg viewBox="0 0 16 16" aria-hidden="true"><use xlink:href="#svg-link"></use></svg></a> Running Individual Modules </h3> <h4 id="just-the-named-allele-matcher"> <a href="#just-the-named-allele-matcher" class="anchor-heading" aria-labelledby="just-the-named-allele-matcher"><svg viewBox="0 0 16 16" aria-hidden="true"><use xlink:href="#svg-link"></use></svg></a> Just the <code class="language-plaintext highlighter-rouge">Named Allele Matcher</code> </h4> <p>This will call the diplotypes for the input VCF file.</p> <p>Examples:</p> <div class="language-console highlighter-rouge"><div class="highlight"><pre class="highlight"><code><span class="gp">#</span><span class="w"> </span>java <span class="nt">-jar</span> pharmcat.jar <span class="nt">-matcher</span> <span class="nt">-vcf</span> /tmp/sample.vcf
<span class="go">Saving named allele matcher JSON results to /tmp/sample.match.json
</span></code></pre></div></div> <h4 id="just-the-phenotyper"> <a href="#just-the-phenotyper" class="anchor-heading" aria-labelledby="just-the-phenotyper"><svg viewBox="0 0 16 16" aria-hidden="true"><use xlink:href="#svg-link"></use></svg></a> Just the <code class="language-plaintext highlighter-rouge">Phenotyper</code> </h4> <p>This will take matcher and/or call data and output function and phenotype information for them.</p> <p>Examples:</p> <div class="language-console highlighter-rouge"><div class="highlight"><pre class="highlight"><code><span class="gp">#</span><span class="w"> </span>java <span class="nt">-jar</span> pharmcat.jar <span class="nt">-phenotyper</span> <span class="nt">-pi</span> /tmp/sample1.match.json
<span class="go">Saving phenotyper JSON results to /tmp/results/sample1.phenotype.json

</span><span class="gp">#</span><span class="w"> </span>java <span class="nt">-jar</span> pharmcat.jar <span class="nt">-phenotyper</span> <span class="nt">-pi</span> /tmp/sample2.match.json <span class="nt">-po</span> /tmp/outside_calls.tsv
<span class="go">Saving phenotyper JSON results to /tmp/results/sample2.phenotype.json

</span><span class="gp">#</span><span class="w"> </span>java <span class="nt">-jar</span> pharmcat.jar <span class="nt">-phenotyper</span> <span class="nt">-po</span> /tmp/outside_calls.tsv
<span class="go">Saving phenotyper JSON results to /tmp/results/outside_calls.phenotype.json
</span></code></pre></div></div> <h4 id="just-the-reporter"> <a href="#just-the-reporter" class="anchor-heading" aria-labelledby="just-the-reporter"><svg viewBox="0 0 16 16" aria-hidden="true"><use xlink:href="#svg-link"></use></svg></a> Just the <code class="language-plaintext highlighter-rouge">Reporter</code> </h4> <p>This will take the phenotyper data and output the relevant drug annotations in a comprehensive HTML report.</p> <p>Examples:</p> <div class="language-console highlighter-rouge"><div class="highlight"><pre class="highlight"><code><span class="gp">#</span><span class="w"> </span>java <span class="nt">-jar</span> pharmcat.jar <span class="nt">-reporter</span> <span class="nt">-ri</span> /tmp/sample1.phenotype.json
<span class="go">Saving reporter HTML results to /tmp/results/sample1.report.html

</span><span class="gp">#</span><span class="w"> </span>java <span class="nt">-jar</span> pharmcat.jar <span class="nt">-reporter</span> <span class="nt">-ri</span> /tmp/sample2.phenotype.json <span class="nt">-rt</span> <span class="s2">"Awesome Report"</span> <span class="nt">-reporterJson</span>
<span class="go">Saving reporter JSON results to /tmp/results/sample1.report.json
Saving reporter HTML results to /tmp/results/sample1.report.html
</span></code></pre></div></div> <h4 id="combinations"> <a href="#combinations" class="anchor-heading" aria-labelledby="combinations"><svg viewBox="0 0 16 16" aria-hidden="true"><use xlink:href="#svg-link"></use></svg></a> Combinations </h4> <p>Run <code class="language-plaintext highlighter-rouge">Named Allele Matcher</code> and <code class="language-plaintext highlighter-rouge">Phenotyper</code>:</p> <div class="language-console highlighter-rouge"><div class="highlight"><pre class="highlight"><code><span class="gp">#</span><span class="w"> </span>java <span class="nt">-jar</span> pharmcat.jar <span class="nt">-matcher</span> <span class="nt">-vcf</span> /tmp/sample1.vcf <span class="nt">-phenotyper</span>
<span class="go">Saving named allele matcher JSON results to /tmp/sample1.match.json
Saving phenotyper JSON results to /tmp/results/sample1.phenotype.json

</span><span class="gp">#</span><span class="w"> </span>java <span class="nt">-jar</span> pharmcat.jar <span class="nt">-matcher</span> <span class="nt">-vcf</span> /tmp/sample2.vcf <span class="nt">-phenotyper</span> <span class="nt">-po</span> /tmp/outside_calls.tsv
<span class="go">Saving named allele matcher JSON results to /tmp/sample2.match.json
Saving phenotyper JSON results to /tmp/results/sample2.phenotype.json
</span></code></pre></div></div> <p>Run <code class="language-plaintext highlighter-rouge">Phenotyper</code> and <code class="language-plaintext highlighter-rouge">Reporter</code>:</p> <div class="language-console highlighter-rouge"><div class="highlight"><pre class="highlight"><code><span class="gp">#</span><span class="w"> </span>java <span class="nt">-jar</span> pharmcat.jar <span class="nt">-phenotyper</span> <span class="nt">-pi</span> /tmp/sample1.phenotype.json <span class="nt">-po</span> /tmp/outside_calls.tsv <span class="nt">-reporter</span>
<span class="go">Saving phenotyper JSON results to /tmp/results/sample1.phenotype.json
Saving reporter HTML results to /tmp/results/sample1.report.html

</span><span class="gp">#</span><span class="w"> </span>java <span class="nt">-jar</span> pharmcat.jar <span class="nt">-phenotyper</span> <span class="nt">-po</span> /tmp/outside_calls.tsv <span class="nt">-reporter</span>
<span class="go">Saving phenotyper JSON results to /tmp/results/outside_calls.phenotype.json
Saving reporter HTML results to /tmp/results/outside_calls.report.html
</span></code></pre></div></div> <h3 id="research-only-options"> <a href="#research-only-options" class="anchor-heading" aria-labelledby="research-only-options"><svg viewBox="0 0 16 16" aria-hidden="true"><use xlink:href="#svg-link"></use></svg></a> Research-Only Options </h3> <p>PharmCAT has a couple experimental options that enables non-standard functionality. These options are non-standard because they're meant for <strong>resarch use only</strong>!</p> <h4 id="calling-combinations-and-partials"> <a href="#calling-combinations-and-partials" class="anchor-heading" aria-labelledby="calling-combinations-and-partials"><svg viewBox="0 0 16 16" aria-hidden="true"><use xlink:href="#svg-link"></use></svg></a> Calling combinations and partials </h4> <p>If given the <code class="language-plaintext highlighter-rouge">--research combinations</code> flag, PharmCAT will try to call combinations and partials. These are only called if an exact match with a named allele cannot be found.</p> <p>A combination is when a sample matches a combination of 2 or more named alleles. For example, CYP2B6 <code class="language-plaintext highlighter-rouge">*6 + *14/*13</code>.</p> <p>A partial is when a sample matches all the core variants of a named allele but also has additional variants. For example, CYP2C19 <code class="language-plaintext highlighter-rouge">"*2/*17 + g.94781859G&gt;A"</code>. The most extreme case of a partial is when it happens off the reference allele. A partial off the reference allele will only be called if the data is phased, or the unphased data only has 2 possible sequence combinations.</p> <h4 id="calling-cyp2d6"> <a href="#calling-cyp2d6" class="anchor-heading" aria-labelledby="calling-cyp2d6"><svg viewBox="0 0 16 16" aria-hidden="true"><use xlink:href="#svg-link"></use></svg></a> Calling CYP2D6 </h4> <p>For details on calling CYP2D6 in PharmCAT, see <a href="/using/Calling-CYP2D6">Calling CYP2D6</a>.</p> <p>This option is not listed here because we do <strong>NOT</strong> recommend calling CYP2D6 from VCF, and <a href="/using/Calling-CYP2D6">Calling CYP2D6</a> explains why and what your options are.</p> <hr> <footer> <div> PharmCAT is managed at Stanford University & University of Pennsylvania (NHGRI U24HG010862). </div> </footer> </div> </div> <div class="search-overlay"></div> </div> </body> </html>
