<!DOCTYPE html> <html lang="en-US"> <head> <meta charset="UTF-8"> <meta http-equiv="X-UA-Compatible" content="IE=Edge"> <title>VCF Preprocessor - PharmCAT</title> <link rel="shortcut icon" href="/favicon.ico" type="image/x-icon"> <link rel="stylesheet" href="/assets/css/just-the-docs-default.css"> <script async src="https://www.googletagmanager.com/gtag/js?id=G-7ZP1HHBWTE"></script> <script> window.dataLayer = window.dataLayer || []; function gtag(){dataLayer.push(arguments);} gtag('js', new Date()); gtag('config', 'G-7ZP1HHBWTE', { 'anonymize_ip': true }); </script> <script type="text/javascript" src="/assets/js/vendor/lunr.min.js"></script> <script type="text/javascript" src="/assets/js/just-the-docs.js"></script> <meta name="viewport" content="width=device-width, initial-scale=1"> <!-- Begin Jekyll SEO tag v2.8.0 --> <title>VCF Preprocessor | PharmCAT</title> <meta name="generator" content="Jekyll v4.2.2" /> <meta property="og:title" content="VCF Preprocessor" /> <meta property="og:locale" content="en_US" /> <meta name="description" content="The Pharmacogenomics Clinical Annotation Tool (PharmCAT) is an open-source tool that takes an individual’s genetic data and determines their respective drug-related genotypes and phenotypes. PharmCAT produces a personalized report with genotype-specific expert-reviewed drug prescribing recommendations." /> <meta property="og:description" content="The Pharmacogenomics Clinical Annotation Tool (PharmCAT) is an open-source tool that takes an individual’s genetic data and determines their respective drug-related genotypes and phenotypes. PharmCAT produces a personalized report with genotype-specific expert-reviewed drug prescribing recommendations." /> <meta property="og:site_name" content="PharmCAT" /> <meta property="og:type" content="website" /> <meta name="twitter:card" content="summary" /> <meta property="twitter:title" content="VCF Preprocessor" /> <script type="application/ld+json"> {"@context":"https://schema.org","@type":"WebPage","description":"The Pharmacogenomics Clinical Annotation Tool (PharmCAT) is an open-source tool that takes an individual’s genetic data and determines their respective drug-related genotypes and phenotypes. PharmCAT produces a personalized report with genotype-specific expert-reviewed drug prescribing recommendations.","headline":"VCF Preprocessor","publisher":{"@type":"Organization","logo":{"@type":"ImageObject","url":"/images/pharmcat_header.png"}},"url":"/using/VCF-Preprocessor.html"}</script> <!-- End Jekyll SEO tag --> <script async defer data-website-id="714c506c-5f1e-4ea8-af79-fa80c02911c6" src="https://metrics.cpicpgx.org/j.js"></script> <style> h4 { font-size: 14px !important; font-weight: bold; text-transform: none; } h4 code { font-size: 0.9em; font-weight: bold; } h5 code { font-size: .9em; } .main-content ul, .main-content ol { padding-left: 2em; } .main-content dl { grid-template-columns: 16em 1fr; } .main-content dt::after { content: unset; } .altArg { font-size: .8em; font-weight: normal; } footer { position: relative; top: -1.5em; font-size: smaller; } </style> </head> <body> <svg xmlns="http://www.w3.org/2000/svg" style="display: none;"> <symbol id="svg-link" viewBox="0 0 24 24"> <title>Link</title> <svg xmlns="http://www.w3.org/2000/svg" width="24" height="24" viewBox="0 0 24 24" fill="none" stroke="currentColor" stroke-width="2" stroke-linecap="round" stroke-linejoin="round" class="feather feather-link"> <path d="M10 13a5 5 0 0 0 7.54.54l3-3a5 5 0 0 0-7.07-7.07l-1.72 1.71"></path><path d="M14 11a5 5 0 0 0-7.54-.54l-3 3a5 5 0 0 0 7.07 7.07l1.71-1.71"></path> </svg> </symbol> <symbol id="svg-search" viewBox="0 0 24 24"> <title>Search</title> <svg xmlns="http://www.w3.org/2000/svg" width="24" height="24" viewBox="0 0 24 24" fill="none" stroke="currentColor" stroke-width="2" stroke-linecap="round" stroke-linejoin="round" class="feather feather-search"> <circle cx="11" cy="11" r="8"></circle><line x1="21" y1="21" x2="16.65" y2="16.65"></line> </svg> </symbol> <symbol id="svg-menu" viewBox="0 0 24 24"> <title>Menu</title> <svg xmlns="http://www.w3.org/2000/svg" width="24" height="24" viewBox="0 0 24 24" fill="none" stroke="currentColor" stroke-width="2" stroke-linecap="round" stroke-linejoin="round" class="feather feather-menu"> <line x1="3" y1="12" x2="21" y2="12"></line><line x1="3" y1="6" x2="21" y2="6"></line><line x1="3" y1="18" x2="21" y2="18"></line> </svg> </symbol> <symbol id="svg-arrow-right" viewBox="0 0 24 24"> <title>Expand</title> <svg xmlns="http://www.w3.org/2000/svg" width="24" height="24" viewBox="0 0 24 24" fill="none" stroke="currentColor" stroke-width="2" stroke-linecap="round" stroke-linejoin="round" class="feather feather-chevron-right"> <polyline points="9 18 15 12 9 6"></polyline> </svg> </symbol> <symbol id="svg-doc" viewBox="0 0 24 24"> <title>Document</title> <svg xmlns="http://www.w3.org/2000/svg" width="24" height="24" viewBox="0 0 24 24" fill="none" stroke="currentColor" stroke-width="2" stroke-linecap="round" stroke-linejoin="round" class="feather feather-file"> <path d="M13 2H6a2 2 0 0 0-2 2v16a2 2 0 0 0 2 2h12a2 2 0 0 0 2-2V9z"></path><polyline points="13 2 13 9 20 9"></polyline> </svg> </symbol> </svg> <div class="side-bar"> <div class="site-header"> <a href="/" class="site-title lh-tight"> <div class="site-logo"></div> </a> <a href="#" id="menu-button" class="site-button"> <svg viewBox="0 0 24 24" class="icon"><use xlink:href="#svg-menu"></use></svg> </a> </div> <nav role="navigation" aria-label="Main" id="site-nav" class="site-nav"> <ul class="nav-list"><li class="nav-list-item"><a href="/" class="nav-list-link">Home</a></li><li class="nav-list-item"><a href="#" class="nav-list-expander"><svg viewBox="0 0 24 24"><use xlink:href="#svg-arrow-right"></use></svg></a><a href="/methods/" class="nav-list-link">Methods</a><ul class="nav-list "><li class="nav-list-item "><a href="/methods/Modules.html" class="nav-list-link">PharmCAT Modules</a></li><li class="nav-list-item "><a href="/methods/determining-alleles" class="nav-list-link">Determining Alleles</a></li><li class="nav-list-item "><a href="/methods/NamedAlleleMatcher-101.html" class="nav-list-link">Named Allele Matcher 101</a></li><li class="nav-list-item "><a href="/methods/NamedAlleleMatcher-201.html" class="nav-list-link">Named Allele Matcher 201</a></li><li class="nav-list-item "><a href="/methods/gene-definition-exceptions" class="nav-list-link">Gene Definition Exceptions</a></li><li class="nav-list-item "><a href="/methods/versioning/" class="nav-list-link">Versioning</a></li></ul></li><li class="nav-list-item active"><a href="#" class="nav-list-expander"><svg viewBox="0 0 24 24"><use xlink:href="#svg-arrow-right"></use></svg></a><a href="/using/" class="nav-list-link">Using PharmCAT</a><ul class="nav-list "><li class="nav-list-item "><a href="/using/Running-PharmCAT.html" class="nav-list-link">Running PharmCAT</a></li><li class="nav-list-item active"><a href="/using/VCF-Preprocessor.html" class="nav-list-link active">VCF Preprocessor</a></li><li class="nav-list-item "><a href="/using/PharmCAT-in-Docker.html" class="nav-list-link">PharmCAT in Docker</a></li><li class="nav-list-item "><a href="/using/Multi-Sample-Analysis.html" class="nav-list-link">Multi-Sample Analysis</a></li><li class="nav-list-item "><a href="/using/Calling-CYP2D6.html" class="nav-list-link">Calling CYP2D6</a></li></ul></li><li class="nav-list-item"><a href="#" class="nav-list-expander"><svg viewBox="0 0 24 24"><use xlink:href="#svg-arrow-right"></use></svg></a><a href="/specifications/" class="nav-list-link">Specifications</a><ul class="nav-list "><li class="nav-list-item "><a href="/specifications/VCF-Requirements.html" class="nav-list-link">VCF Requirements</a></li><li class="nav-list-item "><a href="/specifications/preparing-vcf" class="nav-list-link">Preparing VCF</a></li><li class="nav-list-item "><a href="/specifications/Outside-Call-Format.html" class="nav-list-link">Outside Call Format</a></li></ul></li><li class="nav-list-item"><a href="/genes-drugs/" class="nav-list-link">Genes & Drugs</a></li><li class="nav-list-item"><a href="/examples/" class="nav-list-link">Examples</a></li><li class="nav-list-item"><a href="/faqs/" class="nav-list-link">FAQs</a></li><li class="nav-list-item"><a href="/disclaimers/" class="nav-list-link">Disclaimers and Other Information</a></li><li class="nav-list-item"><a href="/about-us/" class="nav-list-link">About Us</a></li></ul> </nav> <footer class="site-footer"> This site uses <a href="https://github.com/pmarsceill/just-the-docs">Just the Docs</a>, a documentation theme for Jekyll. </footer> </div> <div class="main" id="top"> <div id="main-header" class="main-header"> <div class="search"> <div class="search-input-wrap"> <input type="text" id="search-input" class="search-input" tabindex="0" placeholder="Search PharmCAT" aria-label="Search PharmCAT" autocomplete="off"> <label for="search-input" class="search-label"><svg viewBox="0 0 24 24" class="search-icon"><use xlink:href="#svg-search"></use></svg></label> </div> <div id="search-results" class="search-results"></div> </div> </div> <div id="main-content-wrap" class="main-content-wrap"> <nav aria-label="Breadcrumb" class="breadcrumb-nav"> <ol class="breadcrumb-nav-list"> <li class="breadcrumb-nav-list-item"><a href="/using/">Using PharmCAT</a></li> <li class="breadcrumb-nav-list-item"><span>VCF Preprocessor</span></li> </ol> </nav> <div id="main-content" class="main-content" role="main"> <h1 id="pharmcat-vcf-preprocessor"> <a href="#pharmcat-vcf-preprocessor" class="anchor-heading" aria-labelledby="pharmcat-vcf-preprocessor"><svg viewBox="0 0 16 16" aria-hidden="true"><use xlink:href="#svg-link"></use></svg></a> PharmCAT VCF Preprocessor </h1> <p>The PharmCAT VCF preprocessor is a script that can preprocess VCF files for PharmCAT.</p> <p>This tool will:</p> <ol> <li>Strip out PGx positions that PharmCAT does not care about.</li> <li>Break down a multi-sample VCF to multiple single-sample VCF files.</li> <li>Automatically download the necessary Human Reference Genome Sequence FASTA and index files from the NIH FTP site if they are not provided.</li> <li><strong>Perform VCF normalization</strong> - a standardization process that turns VCF into a parsimonious, left-aligned variant representation format (as discussed in <a href="https://doi.org/10.1093/bioinformatics/btv112">Unified Representation of Genetic Variants</a> by Tan, Abecasis, and Kang).</li> <li>Normalize the multiallelic variant representation to PharmCAT's expectation.</li> <li>Process a subset of samples if a sample file is provided.</li> </ol> <p>The PharmCAT VCF preprocessing produces two types of <strong>output</strong>:</p> <ol> <li>One or more PharmCAT-ready, single-sample VCF file(s)</li> <li>A report of missing pharmacogenomics core allele defining positions in user's input</li> </ol> <h2 id="how-to-run-the-pharmcat-vcf-preprocessing-tool"> <a href="#how-to-run-the-pharmcat-vcf-preprocessing-tool" class="anchor-heading" aria-labelledby="how-to-run-the-pharmcat-vcf-preprocessing-tool"><svg viewBox="0 0 16 16" aria-hidden="true"><use xlink:href="#svg-link"></use></svg></a> How to run the PharmCAT VCF preprocessing tool </h2> <h3 id="prerequisite"> <a href="#prerequisite" class="anchor-heading" aria-labelledby="prerequisite"><svg viewBox="0 0 16 16" aria-hidden="true"><use xlink:href="#svg-link"></use></svg></a> Prerequisite </h3> <p>We assume that the input VCF files are prepared following the <a href="https://samtools.github.io/hts-specs/VCFv4.2.pdf">Variant Call Format (VCF) Version &gt;= 4.1</a>.</p> <p>To run the tool, you need to download the following bioinformatic tools:</p> <ul> <li><a href="http://www.htslib.org/download/">bcftools &gt;= v1.15</a></li> <li><a href="http://www.htslib.org/download/">htslib &gt;= v1.15</a></li> </ul> <p>We assume a working python3 installation with necessary dependencies:</p> <ul> <li>python &gt;= 3.9.4</li> <li>pandas &gt;= 1.4.2</li> <li>scikit-allel &gt;= 1.3.5</li> </ul> <p>To install necessary python packages, run the following code</p> <div class="language-console highlighter-rouge"><div class="highlight"><pre class="highlight"><code><span class="gp">$</span><span class="w"> </span>pip3 <span class="nb">install</span> <span class="nt">-r</span> PharmCAT_VCF_Preprocess_py3_requirements.txt
</code></pre></div></div> <h3 id="command-line"> <a href="#command-line" class="anchor-heading" aria-labelledby="command-line"><svg viewBox="0 0 16 16" aria-hidden="true"><use xlink:href="#svg-link"></use></svg></a> Command line </h3> <p>To normalize and prepare a VCF file (single or multiple samples) for PharmCAT, run the following code substituted with proper arguments/inputs:</p> <div class="language-console highlighter-rouge"><div class="highlight"><pre class="highlight"><code><span class="gp">$</span><span class="w"> </span>python3 PharmCAT_VCF_Preprocess.py <span class="nt">--input_vcf</span> path/to/compressed_vcf.vcf.gz
</code></pre></div></div> <p><strong>Mandatory</strong> argument: <em>either</em> <code class="language-plaintext highlighter-rouge">--input_vcf</code> <em>or</em> <code class="language-plaintext highlighter-rouge">--input_list</code>.</p> <dl> <dt>--input_vcf</dt> <dd>Path to a single VCF file</dd> <dt>--input_list</dt> <dd>Path to file containing list of paths to VCF files (one per line), sorted by chromosome position. All VCF files must have the same set of samples. Use this when data for a sample has been split among multiple files (e.g. VCF files from large cohorts, such as UK Biobank). <p>Example valid <code class="language-plaintext highlighter-rouge">input_list</code> file:</p> <div class="language-plaintext highlighter-rouge"><div class="highlight"><pre class="highlight"><code>chr1_set1.vcf
chr1_set2.vcf
chr2_set1.vcf
chr2_set2.vcf
...
</code></pre></div> </div> <p>Example invalid <code class="language-plaintext highlighter-rouge">input_list</code> file:</p> <div class="language-plaintext highlighter-rouge"><div class="highlight"><pre class="highlight"><code>chr3_set2.vcf
chr2_set2.vcf
chr1_set1.vcf
chr1_set2.vcf
...
</code></pre></div> </div> </dd> </dl> <p>VCF files can have more than 1 sample and should be bgzip compressed. If not bgzip compressed, they will be automatically bgzipped.</p> <p><strong>Optional</strong> arguments:</p> <dl> <dt>--ref_pgx_vcf</dt> <dd>A sorted, compressed VCF of PGx core allele defining positions used by PharmCAT, by default, <code class="language-plaintext highlighter-rouge">pharmcat_positions.vcf.bgz</code> under the current working directory. You can find this VCF in the <code class="language-plaintext highlighter-rouge">pharmcat_preprocessor-&lt;release_version&gt;.tar.gz</code> available from the PharmCAT GitHub releases page.</dd> <dt>--ref_seq</dt> <dd>The <a href="https://www.ncbi.nlm.nih.gov/assembly/GCF_000001405.39/">GRCh38.p13</a> FASTA file. The FASTA file has to be decompressed and indexed (.fai). These mandatory files will be automatically downloaded (~1GB) from the NIH FTP site to the current working directory if not provided by user (see <a href="#notes">Notes</a> for details).</dd> <dt>--sample_file</dt> <dd>The list of samples to be processed and prepared for PharmCAT. The file should contain one sample per line.</dd> <dt>--path_to_bcftools</dt> <dd>Bcftools must be installed. This argument is optional if users can run bcftools directly using the command line <code class="language-plaintext highlighter-rouge">bcftools &lt;commands&gt;</code>. Alternatively, users can download and compile <a href="http://www.htslib.org/download/">bcftools</a> and provide the path to the executable bcftools program as <code class="language-plaintext highlighter-rouge">/path/to/executable/bcftools</code>.</dd> <dt>--path_to_tabix</dt> <dd>Similar to bcftools, tabix must be installed. Tabix is a part of the <a href="http://www.htslib.org/download/">htslib</a>. If users cannot directly run tabix using the command line <code class="language-plaintext highlighter-rouge">tabix &lt;commands&gt;</code>, the alternative is to download and compile <a href="http://www.htslib.org/download/">htslib</a> and provide the path to the executable tabix program as <code class="language-plaintext highlighter-rouge">/path/to/executable/tabix</code> which should be under the htslib program folder.</dd> <dt>--path_to_bgzip</dt> <dd>Similar to tabix, bgzip must be installed. Bgzip is a part of the <a href="http://www.htslib.org/download/">htslib</a>. If users cannot directly run bgzip using the command line <code class="language-plaintext highlighter-rouge">bgzip &lt;commands&gt;</code>, the alternative is to download and compile <a href="http://www.htslib.org/download/">htslib</a> and provide the path to the executable bgzip program as <code class="language-plaintext highlighter-rouge">/path/to/executable/bgzip</code> which should be under the htslib program folder.</dd> <dt>--output_folder</dt> <dd>Output a compressed PharmCAT VCF file to <code class="language-plaintext highlighter-rouge">/path/to/output/pharmcat/vcf</code>. The default is the parent directory of the input.</dd> <dt>--output_prefix</dt> <dd>Prefix of the output VCF files. Default is <code class="language-plaintext highlighter-rouge">pharmcat_ready_vcf</code>.</dd> <dt>--keep_intermediate_files</dt> <dd>This option will help you save useful intermediate files, for example, a normalized, multiallelic VCF named <code class="language-plaintext highlighter-rouge">input_prefix.pgx_regions.normalized.multiallelic.vcf.gz</code>, which will include all PGx regions from the first position to the last one in each chromosome as listed in the reference PGx VCF.</dd> <dt>-0 <span class="altArg"><br />or --missing_to_ref</span></dt> <dd>This option will add missing PGx positions to the output. Missing PGx positions are those whose genotypes are all missing "./." in every single sample. <ul> <li>This option will not convert "./." to "0/0" if any other sample has non-missing genotype at this position as these missing calls are likely missing for good reasons.</li> <li>This <strong>SHOULD ONLY BE USED</strong> if you are sure your data is reference at the missing positions instead of unreadable/uncallable at those positions. Running PharmCAT with positions as missing vs reference can lead to different results.</li> </ul> </dd> </dl> <p><strong>Output</strong></p> <ol> <li>1 or more PharmCAT-ready VCF file(s), which will be named as <code class="language-plaintext highlighter-rouge">output_prefix&gt;_&lt;sample_ID&gt;.vcf</code>, for example, <code class="language-plaintext highlighter-rouge">pharmcat_ready_vcf.sample_1.vcf</code>, <code class="language-plaintext highlighter-rouge">pharmcat_ready_vcf.sample_2.vcf</code>, etc.</li> <li>The report of missing PGx positions, which will be named as <code class="language-plaintext highlighter-rouge">&lt;output_prefix&gt;.missing_pgx_var.vcf.gz</code>, for example <code class="language-plaintext highlighter-rouge">pharmcat_ready_vcf.missing_pgx_var.vcf.gz</code>. This file only reports positions that are missing in all samples.</li> </ol> <p>If <code class="language-plaintext highlighter-rouge">--missing_to_ref</code> is turned on, you can use this report to trace positions whose genotypes are missing in all samples (<code class="language-plaintext highlighter-rouge">./.</code>) in the original input but have now been added into the output VCF(s) as reference (<code class="language-plaintext highlighter-rouge">0/0</code>).</p> <h2 id="tutorial"> <a href="#tutorial" class="anchor-heading" aria-labelledby="tutorial"><svg viewBox="0 0 16 16" aria-hidden="true"><use xlink:href="#svg-link"></use></svg></a> Tutorial </h2> <h3 id="case-1---single-sample-vcf"> <a href="#case-1---single-sample-vcf" class="anchor-heading" aria-labelledby="case-1---single-sample-vcf"><svg viewBox="0 0 16 16" aria-hidden="true"><use xlink:href="#svg-link"></use></svg></a> Case 1 - single-sample VCF </h3> <p>Imagine we have a VCF named <em>"test_1.vcf.gz"</em> to be used in PharmCAT.</p> <div class="language-console highlighter-rouge"><div class="highlight"><pre class="highlight"><code><span class="gp">$</span><span class="w"> </span><span class="nb">gunzip</span> <span class="nt">-c</span> test_1.vcf.gz
<span class="gp">$</span><span class="w"> </span><span class="nb">cat </span>test_1.vcf
<span class="gp">&lt;...header truncated...&gt;</span><span class="w">
</span><span class="gp">#</span>CHROM	POS	ID	REF	ALT	QUAL	FILTER	INFO	FORMAT	Sample_1
<span class="go">2	233760233	rs3064744	C	CAT	.	PASS	.	GT	1/0
2	233760233	rs3064744	CAT	C	.	PASS	.	GT	0/0
2	233760233	rs3064744	C	CATAT	.	PASS	.	GT	0/1
7	117548628	.	GTTTTTTTA	GTTTTTA	.	PASS	.	GT	0/1
</span></code></pre></div></div> <p>Command to run the PharmCAT VCF preprocessor:</p> <div class="language-console highlighter-rouge"><div class="highlight"><pre class="highlight"><code><span class="gp">$</span><span class="w"> </span>python3 PharmCAT_VCF_Preprocess.py <span class="nt">--input_vcf</span> test_1.vcf.gz
</code></pre></div></div> <p>VCF preprocessor will return two files in this test case.</p> <ol> <li>one named <em>"pharmcat_ready_vcf.Sample_1.vcf"</em>, which is a PharmCAT-ready VCF</li> <li>the other named <em>"pharmcat_ready_vcf.missing_pgx_var.vcf.gz"</em> as a report of missing PGx positions.</li> </ol> <p>To be noted, the chr7 variant is not used in PharmCAT and as such, was accordingly removed by the PharmCAT VCF preprocessor.</p> <div class="language-console highlighter-rouge"><div class="highlight"><pre class="highlight"><code><span class="gp">$</span><span class="w"> </span><span class="nb">cat </span>reference.Sample_1.vcf
<span class="gp">&lt;...header truncated...&gt;</span><span class="w">
</span><span class="gp">#</span>CHROM	POS	ID	REF	ALT	QUAL	FILTER	INFO	FORMAT	Sample_1
<span class="go">chr2	233760233	rs3064744	CAT	C,CATATAT,CATAT	.	PASS	PX=UGT1A1	3/2

</span><span class="gp">$</span><span class="w"> </span><span class="nb">gunzip</span> <span class="nt">-c</span> reference.missing_pgx_var.vcf.gz
<span class="gp">$</span><span class="w"> </span><span class="nb">cat </span>reference.missing_pgx_var.vcf
<span class="gp">&lt;...header truncated...&gt;</span><span class="w">
</span><span class="gp">#</span>CHROM	POS	ID	REF	ALT	QUAL	FILTER	INFO	FORMAT	PharmCAT
<span class="go">chr1	97078987	rs114096998	G	T	.	PASS	PX=DPYD	GT	0/0
chr1	97078993	rs148799944	C	G	.	PASS	PX=DPYD	GT	0/0
chr1	97079005	rs140114515	C	T	.	PASS	PX=DPYD	GT	0/0
</span><span class="gp">&lt;...truncated...&gt;</span><span class="w">
</span></code></pre></div></div> <h3 id="case-2---multi-sample-vcf"> <a href="#case-2---multi-sample-vcf" class="anchor-heading" aria-labelledby="case-2---multi-sample-vcf"><svg viewBox="0 0 16 16" aria-hidden="true"><use xlink:href="#svg-link"></use></svg></a> Case 2 - multi-sample VCF </h3> <p>Imagine we have a VCF named <em>"test_2.vcf.gz"</em> that has two samples with different sample names from the case 1.</p> <div class="language-console highlighter-rouge"><div class="highlight"><pre class="highlight"><code><span class="gp">$</span><span class="w"> </span><span class="nb">gunzip</span> <span class="nt">-c</span> example.vcf.gz
<span class="gp">&lt;...header truncated...&gt;</span><span class="w">
</span><span class="gp">#</span>CHROM	POS	ID	REF	ALT	QUAL	FILTER	INFO	FORMAT	Sample_1	Sample_2
<span class="go">1	97740414	rs72549309	AATGA	A	.	PASS	.	GT	1/0	0/1
2	233760233	rs3064744	C	CAT	.	PASS	.	GT	1/0	0/0
2	233760233	rs3064744	CAT	C	.	PASS	.	GT	0/0	0/1
2	233760233	rs3064744	C	CATAT	.	PASS	.	GT	0/1	1/0
7	117548628	.	GTTTTTTTA	GTTTTTA	.	PASS	.	GT	0/1	1/0
10	94942212	rs1304490498	AAGAAATGGAA	A	.	PASS	.	GT	1/0	0/1
13	48037826	rs777311140	G	GCGGG	.	PASS	.	GT	1/0	0/1
19	38499645	rs121918596	GGAG	G	.	PASS	.	GT	1/0	0/1
22	42130727	.	AG	A	.	PASS	.	GT	1/0	0/1
M	1555	.	G	A	PASS	.	GT	1/0	0/1
</span></code></pre></div></div> <p>Command to run the PharmCAT VCF preprocessor:</p> <div class="language-console highlighter-rouge"><div class="highlight"><pre class="highlight"><code><span class="gp">$</span><span class="w"> </span>python3 PharmCAT_VCF_Preprocess.py <span class="nt">--input_vcf</span> test_2.vcf.gz
</code></pre></div></div> <p>VCF preprocessor will return three (3) files in this test case.</p> <ol> <li>one named <em>"pharmcat_ready_vcf.s1.vcf"</em>. Note that the output PharmCAT-ready VCFs will use the exact sample names from the input VCF.</li> <li>one named <em>"pharmcat_ready_vcf.s2.vcf"</em></li> <li>the third named <em>"pharmcat_ready_vcf.missing_pgx_var.vcf.gz"</em> as a report of missing PGx positions.</li> </ol> <div class="language-console highlighter-rouge"><div class="highlight"><pre class="highlight"><code><span class="gp">$</span><span class="w"> </span><span class="nb">cat </span>reference.Sample_1.vcf
<span class="gp">&lt;...header truncated...&gt;</span><span class="w">
</span><span class="gp">#</span>CHROM	POS	ID	REF	ALT	QUAL	FILTER	INFO	FORMAT	Sample_1
<span class="go">chr1    97740410        rs72549309      GATGA   G       .       PASS    PX=DPYD       GT      1/0
chr2    233760233       rs3064744       CAT     C,CATATAT,CATAT .       PASS    PX=UGT1A1 GT      3/2
chr10   94942205        rs1304490498    CAATGGAAAGA     C       .       PASS    PX=CYP2C9     GT      1/0
chr13   48037825        rs777311140     C       CGCGG   .       PASS    PX=NUDT15     GT      1/0
chr19   38499644        rs121918596     TGGA    T       .       PASS    PX=RYR1       GT      1/0

</span><span class="gp">$</span><span class="w"> </span><span class="nb">cat </span>reference.Sample_2.vcf
<span class="gp">&lt;...header truncated...&gt;</span><span class="w">
</span><span class="gp">#</span>CHROM  POS     ID      REF     ALT     QUAL    FILTER  INFO    FORMAT  Sample_2
<span class="go">chr1    97740410        rs72549309      GATGA   G       .       PASS    PX=DPYD       GT      0/1
chr2    233760233       rs3064744       CAT     C,CATATAT,CATAT .       PASS    PX=UGT1A1 GT      2/1
chr10   94942205        rs1304490498    CAATGGAAAGA     C       .       PASS    PX=CYP2C9     GT      0/1
chr13   48037825        rs777311140     C       CGCGG   .       PASS    PX=NUDT15     GT      0/1
chr19   38499644        rs121918596     TGGA    T       .       PASS    PX=RYR1       GT      0/1


</span><span class="gp">$</span><span class="w"> </span><span class="nb">gunzip</span> <span class="nt">-c</span> reference.missing_pgx_var.vcf.gz
<span class="gp">&lt;...header truncated...&gt;</span><span class="w">
</span><span class="gp">#</span>CHROM	POS	ID	REF	ALT	QUAL	FILTER	INFO	FORMAT	PharmCAT
<span class="go">chr1	97078987	rs114096998	G	T	.	PASS	PX=DPYD	GT	0/0
chr1	97078993	rs148799944	C	G	.	PASS	PX=DPYD	GT	0/0
chr1	97079005	rs140114515	C	T	.	PASS	PX=DPYD	GT	0/0
</span><span class="gp">&lt;...truncated...&gt;</span><span class="w">
</span></code></pre></div></div> <h3 id="notes"> <a href="#notes" class="anchor-heading" aria-labelledby="notes"><svg viewBox="0 0 16 16" aria-hidden="true"><use xlink:href="#svg-link"></use></svg></a> Notes </h3> <p>As of Jan 2022, the latest Human Reference Genome assembly is <strong>GRCh38.p13</strong>. It is available through the <a href="https://ftp.ncbi.nlm.nih.gov/genomes/all/GCA/000/001/405/GCA_000001405.15_GRCh38/seqs_for_alignment_pipelines.ucsc_ids/">NCBI RefSeq FTP site</a>. PharmCAT uses the sequence and index files:</p> <ul> <li><a href="https://ftp.ncbi.nlm.nih.gov/genomes/all/GCA/000/001/405/GCA_000001405.15_GRCh38/seqs_for_alignment_pipelines.ucsc_ids/GCA_000001405.15_GRCh38_no_alt_analysis_set.fna.gz">GCA 000001405.15 GRCh38 sequence</a></li> <li><a href="https://ftp.ncbi.nlm.nih.gov/genomes/all/GCA/000/001/405/GCA_000001405.15_GRCh38/seqs_for_alignment_pipelines.ucsc_ids/GCA_000001405.15_GRCh38_no_alt_analysis_set.fna.fai">GCA 000001405.15 GRCh38 index</a></li> </ul> <p>For more information on GRCh38.p13, please refer to the <a href="https://www.ncbi.nlm.nih.gov/assembly/GCF_000001405.39">pertinent NCBI Assembly Website</a>. Download and preparation of the human reference genome followed <a href="http://lh3.github.io/2017/11/13/which-human-reference-genome-to-use">this blog</a>.</p> <hr> <footer> <div> PharmCAT is managed at Stanford University & University of Pennsylvania (NHGRI U24HG010862). </div> </footer> </div> </div> <div class="search-overlay"></div> </div> </body> </html>
